
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
        <meta name="author" content="Authors">
      
      
      
        <link rel="prev" href="../../cancer-module-sv/sv_tut/">
      
      
        <link rel="next" href="../../cancer-module-somatic/01_signatures/">
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.2">
    
    
      
        <title>Variant Visualisation - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.f56500e0.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#key-learning-outcomes" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-header__button md-logo" aria-label="Markdown Manuals for Bioinformatics Training Platform (BTP)" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Markdown Manuals for Bioinformatics Training Platform (BTP)
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Variant Visualisation
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-nav__button md-logo" aria-label="Markdown Manuals for Bioinformatics Training Platform (BTP)" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../trainers/trainers/" class="md-nav__link">
        The Trainers
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          Timetable
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Timetable" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Timetable
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_introNGS/" class="md-nav__link">
        Intro to NGS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_cancer/" class="md-nav__link">
        Cancer Genomics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_longRead/" class="md-nav__link">
        Long Read
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../preamble/" class="md-nav__link">
        Workshop Information
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Introduction to NGS Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Introduction to NGS Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Introduction to NGS Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-cli/commandline/" class="md-nav__link">
        Introduction to Command Line
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-qc/ngs-qc/" class="md-nav__link">
        Data Quality
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-mapping/alignment/" class="md-nav__link">
        Read Alignment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-chip-seq/chip-seq/" class="md-nav__link">
        ChIP-seq
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-rna-seq/rna-seq/" class="md-nav__link">
        RNA-Seq
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-bio/biol/" class="md-nav__link">
        NGS Biological Insight
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-velvet/velvet/" class="md-nav__link">
        de novo Genome Assembly
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-denovo-canu/denovo_canu/" class="md-nav__link">
        DeNovo Canu
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_6">
          Cancer Genomics Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Cancer Genomics Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Cancer Genomics Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-alignment/alignment/" class="md-nav__link">
        Read Alignment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-snv/snv/" class="md-nav__link">
        Single Nucleotide Variant Calling and Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-cnv/cnv-tut/" class="md-nav__link">
        Copy Number Variation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-sv/sv_tut/" class="md-nav__link">
        Structural Variant Analysis
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Variant Visualisation
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Variant Visualisation
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav" aria-label="Resources You’ll be Using">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#author-information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#acknowledgements" class="md-nav__link">
    Acknowledgements
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#license" class="md-nav__link">
    License
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6_6" type="checkbox" id="__nav_6_6" >
      
      
      
      
        <label class="md-nav__link" for="__nav_6_6">
          Somatic Mutations
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Somatic Mutations" data-md-level="2">
        <label class="md-nav__title" for="__nav_6_6">
          <span class="md-nav__icon md-icon"></span>
          Somatic Mutations
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-somatic/01_signatures/" class="md-nav__link">
        Assessing somatic mutational signatures
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-somatic/02_intogen/" class="md-nav__link">
        Cohort analysis for the identification of driver genes
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_7" type="checkbox" id="__nav_7" >
      
      
      
      
        <label class="md-nav__link" for="__nav_7">
          DeNovo Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="DeNovo Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          DeNovo Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../denovo-module-bac/bac_asm/" class="md-nav__link">
        Bacterial Assembly
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../denovo-module-euk/eukaryotic_assem/" class="md-nav__link">
        Eukaryote Assembly
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../denovo-module-scaffolding/scaffold/" class="md-nav__link">
        Scaffolding
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_8" type="checkbox" id="__nav_8" >
      
      
      
      
        <label class="md-nav__link" for="__nav_8">
          Metagenomics Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Metagenomics Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_8">
          <span class="md-nav__icon md-icon"></span>
          Metagenomics Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-qc/ngs-qc/" class="md-nav__link">
        Metagenomics QC
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-tax/tax/" class="md-nav__link">
        Metagenomics Taxonomic
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-vis/vis/" class="md-nav__link">
        Metagenomics Visualisation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-wga/wga/" class="md-nav__link">
        Metagenomics WGA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-fda/fda/" class="md-nav__link">
        Metagenomics Functional Analysis
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav" aria-label="Resources You’ll be Using">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#author-information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#acknowledgements" class="md-nav__link">
    Acknowledgements
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#license" class="md-nav__link">
    License
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



  <h1>Variant Visualisation</h1>

<p>This module was written by Mathieu Bourgey and the original on-line version is available
<a href="https://github.com/mbourgey/EBI_cancer_workshop_visualization/tree/australia_2015">here.</a></p>
<hr />
<h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<p>After completing this practical the trainee should be able to:</p>
<ul>
<li>Generate Circos like graphics using R</li>
</ul>
<hr />
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>R:<br />
<a href="https://cran.r-project.org/">https://cran.r-project.org/</a></p>
<p>R package circlize:<br />
<a href="https://cran.r-project.org/web/packages/circlize/index.html">https://cran.r-project.org/web/packages/circlize/index.html</a></p>
<hr />
<h2 id="author-information">Author Information<a class="headerlink" href="#author-information" title="Permanent link">&para;</a></h2>
<p><em>Primary Author(s):</em><br />
Mathieu Bourgey <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#109;&#97;&#116;&#104;&#105;&#101;&#117;&#46;&#98;&#111;&#117;&#114;&#103;&#101;&#121;&#64;&#109;&#99;&#103;&#105;&#108;&#108;&#46;&#99;&#97;">&#109;&#97;&#116;&#104;&#105;&#101;&#117;&#46;&#98;&#111;&#117;&#114;&#103;&#101;&#121;&#64;&#109;&#99;&#103;&#105;&#108;&#108;&#46;&#99;&#97;</a>  </p>
<p><em>Contributor(s):</em>    </p>
<hr />
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<p>This short workshop will show you how to visualize your data.</p>
<p>We will be working on 3 types of somatic calls:</p>
<ul>
<li>
<p>SNV calls from MuTect (vcf)</p>
</li>
<li>
<p>SV calls from DELLY (vcf)</p>
</li>
<li>
<p>CNV calls from SCoNEs (tsv)</p>
</li>
</ul>
<hr />
<h2 id="prepare-the-environment">Prepare the Environment<a class="headerlink" href="#prepare-the-environment" title="Permanent link">&para;</a></h2>
<p>We will use a dataset derived from the analysis of whole genome sequencing
paired normal/tumour samples.</p>
<p>The call files are contained in the folder <code>visualization</code>:</p>
<ul>
<li><code>mutect.somatic.vcf</code>  </li>
<li><code>delly.somatic.vcf</code>    </li>
<li><code>scones.somatic.tsv</code>   </li>
</ul>
<p>Many tools are available to do this and the most commonly known is Circos.
Circos is a really not user-friendly. In this tutorial, we show you an
easy alternative to build a circular representation of genomic data.</p>
<p>First we need to go in the folder <code>visualization</code> to do the analysis:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>cd /home/trainee/visualization/
</code></pre></div></td></tr></table></div>

<p>Let&rsquo;s see what is in this folder:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>ls
</code></pre></div></td></tr></table></div>

<p><code>circos.R  delly.somatic.vcf  mutect.somatic.vcf  scones.somatic.tsv</code></p>
<p>Take a look at the data files.</p>
<p>This data has <em>not</em> been restricted to a short piece of a chromosome and SNVs
have already been filtered:</p>
<p><div class="highlight"><pre><span></span><code>less mutect.somatic.tsv
less delly.somatic.vcf
less data/scones.somatic.30k.tsv
</code></pre></div>
<br></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What can you see from this data?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<ul>
<li>
<p>the filtered output of 3 different software: Mutect (SNVs),
Delly (SVs), SCoNEs (CNVs).</p>
</li>
<li>
<p>The 3 files show 2 different formats (vcf, tsv).</p>
</li>
<li>
<p>Almost all type of variants are represented here: mutations,
deletion, inversion, translocation, large amplification and deletion
(CNVs).</p>
</li>
</ul>
</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why don’t we use the vcf format for all types of calls?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>The 1000 Genomes project tries to use/include SV calls in the vcf format.
Some tools like Delly use this format for SVs. It is a good idea to try to
include everything together but this not the be the best way to handle SVs and
CNVs.</p>
</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>Due to the nature of these calls, you can not easily integrate the
positional information of the two breakpoints (that could be located
far away or in an other chromosome) using a single position format.</p>
</details>
</div>
</div>
<p><br></p>
<p>The analysis will be done using the R program:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>R
</code></pre></div></td></tr></table></div>

<p>We will use the circlize package from the cran R project. This package generates
circular plots and has the advantage of being able to provide
pre-built functions for genomic data. One of the main advantages of this
tool is the use of bed format as input data.</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>library(circlize)
</code></pre></div></td></tr></table></div>

<p><br>
Let’s import the variants:</p>
<div class="highlight"><pre><span></span><code><span class="n">snp</span><span class="o">=</span><span class="nf">read.table</span><span class="p">(</span><span class="s">&quot;mutect.somatic.vcf&quot;</span><span class="p">)</span>
<span class="n">sv</span><span class="o">=</span><span class="nf">read.table</span><span class="p">(</span><span class="s">&quot;somatic.sv.vcf&quot;</span><span class="p">)</span>
<span class="n">cnv</span><span class="o">=</span><span class="nf">read.table</span><span class="p">(</span><span class="s">&quot;data/scones.somatic.tsv&quot;</span><span class="p">,</span><span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span>
</code></pre></div>
<p><br>
We need to set up the generic graphical parameters:</p>
<div class="highlight"><pre><span></span><code><span class="nf">par</span><span class="p">(</span><span class="n">mar</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">))</span>
<span class="nf">circos.par</span><span class="p">(</span><span class="s">&quot;start.degree&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">)</span>
<span class="nf">circos.par</span><span class="p">(</span><span class="s">&quot;track.height&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span>
<span class="nf">circos.par</span><span class="p">(</span><span class="s">&quot;canvas.xlim&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">-1.3</span><span class="p">,</span><span class="w"> </span><span class="m">1.3</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;canvas.ylim&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">-1.3</span><span class="p">,</span><span class="w"> </span><span class="m">1.3</span><span class="p">))</span>
</code></pre></div>
<p><br>
Let’s draw hg19 reference ideograms:</p>
<div class="highlight"><pre><span></span><code><span class="nf">circos.initializeWithIdeogram</span><span class="p">(</span><span class="n">species</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;hg19&quot;</span><span class="p">)</span>
</code></pre></div>
<p><br>
Unfortunately circlize does not support hg38 yet. So we will need to
reformat our data to fit the hg19 standards. As we are working only on autosomes,
we won’t need to lift-over and we could simply add <em>chr</em> at the beginning
of the chromosome names.</p>
<p>We can now draw 1 track for somatic mutations:</p>
<div class="highlight"><pre><span></span><code><span class="n">snv_tmp</span><span class="o">=</span><span class="nf">read.table</span><span class="p">(</span><span class="s">&quot;data/mutec.somatic.vcf&quot;</span><span class="p">,</span><span class="n">comment.char</span><span class="o">=</span><span class="s">&quot;#&quot;</span><span class="p">)</span>
<span class="n">snv</span><span class="o">=</span><span class="nf">cbind</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="nf">as.character</span><span class="p">(</span><span class="n">snp</span><span class="p">[,</span><span class="m">1</span><span class="p">]),</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="n">snp</span><span class="p">[</span><span class="m">2</span><span class="p">],</span><span class="n">snp</span><span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="m">+1</span><span class="p">)</span>
<span class="nf">circos.genomicTrackPlotRegion</span><span class="p">(</span><span class="n">snv</span><span class="p">,</span><span class="n">stack</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">panel.fun</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">region</span><span class="p">,</span><span class="w"> </span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="kc">...</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nf">circos.genomicPoints</span><span class="p">(</span><span class="n">region</span><span class="p">,</span><span class="w"> </span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span><span class="w"> </span><span class="n">pch</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">9</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;orange&#39;</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="kc">...</span><span class="p">)</span>
<span class="p">})</span>
</code></pre></div>
<p><br>
Let’s draw the 2 tracks for CNVs. One track for duplications in red and
one blue track for deletions.</p>
<div class="highlight"><pre><span></span><code><span class="n">dup</span><span class="o">=</span><span class="n">cnv</span><span class="p">[</span><span class="n">cnv</span><span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="o">&gt;</span><span class="m">2</span><span class="p">,]</span>
<span class="n">dup</span><span class="p">[,</span><span class="m">1</span><span class="p">]</span><span class="o">=</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="nf">as.character</span><span class="p">(</span><span class="n">dup</span><span class="p">[,</span><span class="m">1</span><span class="p">]),</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">)</span>
<span class="n">del</span><span class="o">=</span><span class="n">cnv</span><span class="p">[</span><span class="n">cnv</span><span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="o">&lt;</span><span class="m">2</span><span class="p">,]</span>
<span class="n">del</span><span class="p">[,</span><span class="m">1</span><span class="p">]</span><span class="o">=</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="nf">as.character</span><span class="p">(</span><span class="n">del</span><span class="p">[,</span><span class="m">1</span><span class="p">]),</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">)</span>
<span class="nf">circos.genomicTrackPlotRegion</span><span class="p">(</span><span class="n">dup</span><span class="p">,</span><span class="w"> </span><span class="n">stack</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="n">panel.fun</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">region</span><span class="p">,</span><span class="w"> </span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="kc">...</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span><span class="nf">circos.genomicRect</span><span class="p">(</span><span class="n">region</span><span class="p">,</span><span class="w"> </span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="n">bg.border</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">cex</span><span class="o">=</span><span class="m">1</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="kc">...</span><span class="p">)</span>
<span class="p">})</span>
<span class="nf">circos.genomicTrackPlotRegion</span><span class="p">(</span><span class="n">del</span><span class="p">,</span><span class="w"> </span><span class="n">stack</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="n">panel.fun</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">region</span><span class="p">,</span><span class="w"> </span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="kc">...</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span><span class="nf">circos.genomicRect</span><span class="p">(</span><span class="n">region</span><span class="p">,</span><span class="w"> </span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="n">bg.border</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">cex</span><span class="o">=</span><span class="m">1</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="kc">...</span><span class="p">)</span>
<span class="p">})</span>
</code></pre></div>
<p>We can clearly see a massive deletion in chromosome 3.</p>
<p><br>
To finish we just need to draw 3 tracks + positional links to represent
SVs.</p>
<p>Unfortunately the vcf format has not been designed for SVs. SVs are
defined by 2 breakpoints and the vcf format stores the second one in the
info field. So we will need to extract this information to draw these
calls.</p>
<div class="highlight"><pre><span></span><code><span class="n">chrEnd</span><span class="o">=</span><span class="kc">NULL</span>
<span class="n">posEnd</span><span class="o">=</span><span class="kc">NULL</span>
<span class="nf">for </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">dim</span><span class="p">(</span><span class="n">sv</span><span class="p">)[</span><span class="m">1</span><span class="p">])</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">addInfo</span><span class="o">=</span><span class="nf">strsplit</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="n">sv</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="m">8</span><span class="p">]),</span><span class="n">split</span><span class="o">=</span><span class="s">&quot;;&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="n">chrInf</span><span class="o">=</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">addInfo</span><span class="p">[[</span><span class="m">1</span><span class="p">]][</span><span class="m">3</span><span class="p">],</span><span class="n">split</span><span class="o">=</span><span class="s">&quot;=&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="n">chrEnd</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="n">chrEnd</span><span class="p">,</span><span class="n">chrInf</span><span class="p">[[</span><span class="m">1</span><span class="p">]][</span><span class="m">2</span><span class="p">])</span>
<span class="w">    </span><span class="n">posInf</span><span class="o">=</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">addInfo</span><span class="p">[[</span><span class="m">1</span><span class="p">]][</span><span class="m">4</span><span class="p">],</span><span class="n">split</span><span class="o">=</span><span class="s">&quot;=&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="n">posEnd</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="n">posEnd</span><span class="p">,</span><span class="n">posInf</span><span class="p">[[</span><span class="m">1</span><span class="p">]][</span><span class="m">2</span><span class="p">])</span>
<span class="p">}</span>
<span class="n">svTable</span><span class="o">=</span><span class="nf">data.frame</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="n">sv</span><span class="p">[,</span><span class="m">1</span><span class="p">],</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">sv</span><span class="p">[,</span><span class="m">2</span><span class="p">]),</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">posEnd</span><span class="p">),</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="n">chrEnd</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="nf">as.character</span><span class="p">(</span><span class="n">sv</span><span class="p">[,</span><span class="m">5</span><span class="p">]))</span>
</code></pre></div>
<p><br>
Now that we have reformatted the SV calls, let’s draw them.</p>
<div class="highlight"><pre><span></span><code><span class="n">typeE</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;&lt;DEL&gt;&quot;</span><span class="p">,</span><span class="s">&quot;&lt;INS&gt;&quot;</span><span class="p">,</span><span class="s">&quot;&lt;INV&gt;&quot;</span><span class="p">)</span>
<span class="n">colE</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="s">&quot;green&quot;</span><span class="p">)</span>
<span class="nf">for </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span><span class="n">bed_list</span><span class="o">=</span><span class="n">svTable</span><span class="p">[</span><span class="n">svTable</span><span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="o">==</span><span class="n">typeE</span><span class="p">[</span><span class="n">i</span><span class="p">],]</span>
<span class="w">        </span><span class="nf">circos.genomicTrackPlotRegion</span><span class="p">(</span><span class="n">bed_list</span><span class="p">,</span><span class="n">stack</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">panel.fun</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">region</span><span class="p">,</span><span class="w"> </span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="kc">...</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">                </span><span class="nf">circos.genomicPoints</span><span class="p">(</span><span class="n">region</span><span class="p">,</span><span class="w"> </span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">pch</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">16</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colE</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="kc">...</span><span class="p">)</span>
<span class="w">        </span><span class="p">})</span>
<span class="p">}</span>

<span class="n">bed1</span><span class="o">=</span><span class="nf">cbind</span><span class="p">(</span><span class="n">svTable</span><span class="p">[</span><span class="n">svTable</span><span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="o">==</span><span class="s">&quot;&lt;TRA&gt;&quot;</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">],</span><span class="n">svTable</span><span class="p">[</span><span class="n">svTable</span><span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="o">==</span><span class="s">&quot;&lt;TRA&gt;&quot;</span><span class="p">,</span><span class="m">2</span><span class="p">]</span><span class="m">+5</span><span class="p">)</span>
<span class="n">bed2</span><span class="o">=</span><span class="nf">cbind</span><span class="p">(</span><span class="n">svTable</span><span class="p">[</span><span class="n">svTable</span><span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="o">==</span><span class="s">&quot;&lt;TRA&gt;&quot;</span><span class="p">,</span><span class="nf">c</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="m">3</span><span class="p">)],</span><span class="n">svTable</span><span class="p">[</span><span class="n">svTable</span><span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="o">==</span><span class="s">&quot;&lt;TRA&gt;&quot;</span><span class="p">,</span><span class="m">3</span><span class="p">]</span><span class="m">+5</span><span class="p">)</span>

<span class="nf">for </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">dim</span><span class="p">(</span><span class="n">bed1</span><span class="p">)[</span><span class="m">1</span><span class="p">])</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nf">circos.link</span><span class="p">(</span><span class="n">bed1</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="m">1</span><span class="p">],</span><span class="n">bed1</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="m">2</span><span class="p">],</span><span class="n">bed2</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="m">1</span><span class="p">],</span><span class="n">bed2</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="m">2</span><span class="p">])</span>
<span class="p">}</span>
</code></pre></div>
<p><br>
A good graph needs a title and legend:</p>
<div class="highlight"><pre><span></span><code><span class="nf">title</span><span class="p">(</span><span class="s">&quot;Somatic calls (SNV - SV - CNV)&quot;</span><span class="p">)</span>
<span class="nf">legend</span><span class="p">(</span><span class="m">0.7</span><span class="p">,</span><span class="m">1.4</span><span class="p">,</span><span class="n">legend</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;SNV&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;CNV-DUPLICATION&quot;</span><span class="p">,</span><span class="s">&quot;CNV-DELETION&quot;</span><span class="p">,</span><span class="s">&quot;SV-DELETION&quot;</span><span class="p">,</span><span class="s">&quot;SV-INSERTION&quot;</span><span class="p">,</span><span class="s">&quot;SV-INVERSION&quot;</span><span class="p">),</span><span class="n">col</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;orange&quot;</span><span class="p">,</span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="s">&quot;green&quot;</span><span class="p">,</span><span class="s">&quot;red&quot;</span><span class="p">),</span><span class="n">pch</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">16</span><span class="p">,</span><span class="m">15</span><span class="p">,</span><span class="m">15</span><span class="p">,</span><span class="m">16</span><span class="p">,</span><span class="m">16</span><span class="p">,</span><span class="m">16</span><span class="p">,</span><span class="m">16</span><span class="p">,</span><span class="m">16</span><span class="p">),</span><span class="n">cex</span><span class="o">=</span><span class="m">0.75</span><span class="p">,</span><span class="n">title</span><span class="o">=</span><span class="s">&quot;Tracks:&quot;</span><span class="p">,</span><span class="n">bty</span><span class="o">=</span><span class="s">&#39;n&#39;</span><span class="p">)</span>
<span class="nf">legend</span><span class="p">(</span><span class="m">0.6</span><span class="p">,</span><span class="m">0.95</span><span class="p">,</span><span class="n">legend</span><span class="o">=</span><span class="s">&quot;SV-TRANSLOCATION&quot;</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="n">lty</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="n">cex</span><span class="o">=</span><span class="m">0.75</span><span class="p">,</span><span class="n">lwd</span><span class="o">=</span><span class="m">1.2</span><span class="p">,</span><span class="n">bty</span><span class="o">=</span><span class="s">&#39;n&#39;</span><span class="p">)</span>
</code></pre></div>
<p><br>
You should obtain a plot like this one:</p>
<p><img alt="image" src="../images/circos.png" /></p>
<p><br></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Could you generate the graph and save it into a pdf file?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<div class="highlight"><pre><span></span><code><span class="nf">circos.clear</span><span class="p">()</span>
<span class="nf">pdf</span><span class="p">(</span><span class="s">&quot;circos.pdf&quot;</span><span class="p">)</span>
<span class="kc">...</span>
<span class="nf">dev.off</span><span class="p">()</span>
</code></pre></div>
</details>
</div>
</div>
<p>Finally exit R
  <div class="highlight"><pre><span></span><code><span class="nf">q</span><span class="p">(</span><span class="s">&quot;yes&quot;</span><span class="p">)</span>
</code></pre></div></p>
<hr />
<h2 id="acknowledgements">Acknowledgements<a class="headerlink" href="#acknowledgements" title="Permanent link">&para;</a></h2>
<p>Mathieu Bourgey would like to thank and acknowledge Louis Letourneau for this help and
for sharing his material. The format of the tutorial has been inspired
from Mar Gonzalez Porta. I also want to acknowledge Joel Fillon, Louis
Letrouneau (again), Francois Lefebvre, Maxime Caron and Guillaume
Bourque for the help in building these pipelines and working with all
the various datasets.</p>
<h2 id="license">License<a class="headerlink" href="#license" title="Permanent link">&para;</a></h2>
<p>This work is licensed under a Creative Commons Attribution-ShareAlike
3.0 Unported License. This means that you are able to copy, share and
modify the work, as long as the result is distributed under the same
license.</p>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      &copy; The authors; 2017-
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../../..", "features": [], "search": "../../../assets/javascripts/workers/search.12658920.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.5cf534bf.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
  </body>
</html>