
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
        <meta name="author" content="Authors">
      
      
      
        <link rel="prev" href="../../cancer-module-somatic/02_intogen/">
      
      
        <link rel="next" href="../../denovo-module-euk/eukaryotic_assem/">
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.2">
    
    
      
        <title>Bacterial Assembly - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.f56500e0.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#key-learning-outcomes" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-header__button md-logo" aria-label="Markdown Manuals for Bioinformatics Training Platform (BTP)" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Markdown Manuals for Bioinformatics Training Platform (BTP)
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Bacterial Assembly
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-nav__button md-logo" aria-label="Markdown Manuals for Bioinformatics Training Platform (BTP)" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../trainers/trainers/" class="md-nav__link">
        The Trainers
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          Timetable
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Timetable" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Timetable
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_introNGS/" class="md-nav__link">
        Intro to NGS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_cancer/" class="md-nav__link">
        Cancer Genomics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_longRead/" class="md-nav__link">
        Long Read
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../preamble/" class="md-nav__link">
        Workshop Information
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Introduction to NGS Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Introduction to NGS Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Introduction to NGS Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-cli/commandline/" class="md-nav__link">
        Introduction to Command Line
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-qc/ngs-qc/" class="md-nav__link">
        Data Quality
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-mapping/alignment/" class="md-nav__link">
        Read Alignment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-chip-seq/chip-seq/" class="md-nav__link">
        ChIP-seq
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-rna-seq/rna-seq/" class="md-nav__link">
        RNA-Seq
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-bio/biol/" class="md-nav__link">
        NGS Biological Insight
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-velvet/velvet/" class="md-nav__link">
        de novo Genome Assembly
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-denovo-canu/denovo_canu/" class="md-nav__link">
        DeNovo Canu
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" >
      
      
      
      
        <label class="md-nav__link" for="__nav_6">
          Cancer Genomics Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Cancer Genomics Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Cancer Genomics Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-alignment/alignment/" class="md-nav__link">
        Read Alignment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-snv/snv/" class="md-nav__link">
        Single Nucleotide Variant Calling and Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-cnv/cnv-tut/" class="md-nav__link">
        Copy Number Variation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-sv/sv_tut/" class="md-nav__link">
        Structural Variant Analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-viz/visu/" class="md-nav__link">
        Variant Visualisation
      </a>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6_6" type="checkbox" id="__nav_6_6" >
      
      
      
      
        <label class="md-nav__link" for="__nav_6_6">
          Somatic Mutations
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Somatic Mutations" data-md-level="2">
        <label class="md-nav__title" for="__nav_6_6">
          <span class="md-nav__icon md-icon"></span>
          Somatic Mutations
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-somatic/01_signatures/" class="md-nav__link">
        Assessing somatic mutational signatures
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-somatic/02_intogen/" class="md-nav__link">
        Cohort analysis for the identification of driver genes
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_7" type="checkbox" id="__nav_7" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_7">
          DeNovo Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="DeNovo Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          DeNovo Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Bacterial Assembly
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Bacterial Assembly
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav" aria-label="Resources You’ll be Using">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-source-and-useful-links" class="md-nav__link">
    Data Source and Useful Links
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#author-information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#before-the-assembly" class="md-nav__link">
    Before the assembly
  </a>
  
    <nav class="md-nav" aria-label="Before the assembly">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sequence-data" class="md-nav__link">
    Sequence data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trim-or-clip-reads" class="md-nav__link">
    Trim or clip reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#allocate-yourself-to-a-k-value-on-the-spreadsheet" class="md-nav__link">
    Allocate yourself to a K value on the spreadsheet
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assemble-the-reads-using-velvet" class="md-nav__link">
    Assemble the reads using Velvet
  </a>
  
    <nav class="md-nav" aria-label="Assemble the reads using Velvet">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#add-your-results-to-the-spreadsheet" class="md-nav__link">
    Add your results to the spreadsheet
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#effect-of-k-on-assembly" class="md-nav__link">
    Effect of K on assembly
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#output-files" class="md-nav__link">
    Output Files
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#visualising-the-assembly-graph-using-bandage" class="md-nav__link">
    Visualising the assembly graph using Bandage
  </a>
  
    <nav class="md-nav" aria-label="Visualising the assembly graph using Bandage">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#explore-the-graph" class="md-nav__link">
    Explore the graph
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#features-of-the-assembly-graph" class="md-nav__link">
    Features of the assembly graph
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#conclusion" class="md-nav__link">
    Conclusion
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../denovo-module-euk/eukaryotic_assem/" class="md-nav__link">
        Eukaryote Assembly
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../denovo-module-scaffolding/scaffold/" class="md-nav__link">
        Scaffolding
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_8" type="checkbox" id="__nav_8" >
      
      
      
      
        <label class="md-nav__link" for="__nav_8">
          Metagenomics Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Metagenomics Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_8">
          <span class="md-nav__icon md-icon"></span>
          Metagenomics Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-qc/ngs-qc/" class="md-nav__link">
        Metagenomics QC
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-tax/tax/" class="md-nav__link">
        Metagenomics Taxonomic
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-vis/vis/" class="md-nav__link">
        Metagenomics Visualisation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-wga/wga/" class="md-nav__link">
        Metagenomics WGA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-fda/fda/" class="md-nav__link">
        Metagenomics Functional Analysis
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav" aria-label="Resources You’ll be Using">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-source-and-useful-links" class="md-nav__link">
    Data Source and Useful Links
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#author-information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#before-the-assembly" class="md-nav__link">
    Before the assembly
  </a>
  
    <nav class="md-nav" aria-label="Before the assembly">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sequence-data" class="md-nav__link">
    Sequence data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trim-or-clip-reads" class="md-nav__link">
    Trim or clip reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#allocate-yourself-to-a-k-value-on-the-spreadsheet" class="md-nav__link">
    Allocate yourself to a K value on the spreadsheet
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assemble-the-reads-using-velvet" class="md-nav__link">
    Assemble the reads using Velvet
  </a>
  
    <nav class="md-nav" aria-label="Assemble the reads using Velvet">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#add-your-results-to-the-spreadsheet" class="md-nav__link">
    Add your results to the spreadsheet
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#effect-of-k-on-assembly" class="md-nav__link">
    Effect of K on assembly
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#output-files" class="md-nav__link">
    Output Files
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#visualising-the-assembly-graph-using-bandage" class="md-nav__link">
    Visualising the assembly graph using Bandage
  </a>
  
    <nav class="md-nav" aria-label="Visualising the assembly graph using Bandage">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#explore-the-graph" class="md-nav__link">
    Explore the graph
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#features-of-the-assembly-graph" class="md-nav__link">
    Features of the assembly graph
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#conclusion" class="md-nav__link">
    Conclusion
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



  <h1>Bacterial Assembly</h1>

<h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<p>After completing this module the trainee should be able to:</p>
<ul>
<li>
<p>Perform a simple genome assembly for a small organism using <code>Velvet</code></p>
</li>
<li>
<p>Visualise the assembly graph using <code>Bandage</code></p>
</li>
<li>
<p>Be aware of the effects and trade-offs of the parameter <code>K</code> on the
    genome assembly</p>
</li>
<li>
<p>Understand that genome assembly produces a graph structure</p>
</li>
</ul>
<hr />
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>Velvet:<br />
<a href="https://www.ebi.ac.uk/~zerbino/velvet/">https://www.ebi.ac.uk/~zerbino/velvet/</a></p>
<p>Bandage:<br />
<a href="https://rrwick.github.io/Bandage/">https://rrwick.github.io/Bandage/</a></p>
<hr />
<h2 id="data-source-and-useful-links">Data Source and Useful Links<a class="headerlink" href="#data-source-and-useful-links" title="Permanent link">&para;</a></h2>
<p>Data used in this tutorial is from ENA accession SRR2054105:  <br />
<a href="http://www.ebi.ac.uk/ena/data/view/SRR2054105">http://www.ebi.ac.uk/ena/data/view/SRR2054105</a></p>
<p>Spreadsheet used in the group activity:<br />
<a href="https://docs.google.com/spreadsheets/d/1iFbCCihawpY1LClsAB-OJ66lyeW7EsdJyaMo1HCetF8/edit?usp=sharing">https://docs.google.com/spreadsheets/d/1iFbCCihawpY1LClsAB-OJ66lyeW7EsdJyaMo1HCetF8/edit?usp=sharing</a></p>
<hr />
<h2 id="author-information">Author Information<a class="headerlink" href="#author-information" title="Permanent link">&para;</a></h2>
<p><em>Primary Author(s):</em><br />
Torsten Seemann <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#116;&#115;&#101;&#101;&#109;&#97;&#110;&#110;&#64;&#117;&#110;&#105;&#109;&#101;&#108;&#98;&#46;&#101;&#100;&#117;&#46;&#97;&#117;">&#116;&#115;&#101;&#101;&#109;&#97;&#110;&#110;&#64;&#117;&#110;&#105;&#109;&#101;&#108;&#98;&#46;&#101;&#100;&#117;&#46;&#97;&#117;</a><br />
Paul Berkman <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#80;&#97;&#117;&#108;&#46;&#66;&#101;&#114;&#107;&#109;&#97;&#110;&#64;&#99;&#115;&#105;&#114;&#111;&#46;&#97;&#117;">&#80;&#97;&#117;&#108;&#46;&#66;&#101;&#114;&#107;&#109;&#97;&#110;&#64;&#99;&#115;&#105;&#114;&#111;&#46;&#97;&#117;</a><br />
Philippe Moncuquet <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#80;&#104;&#105;&#108;&#105;&#112;&#112;&#101;&#46;&#77;&#111;&#110;&#99;&#117;&#113;&#117;&#101;&#116;&#64;&#99;&#115;&#105;&#114;&#111;&#46;&#97;&#117;">&#80;&#104;&#105;&#108;&#105;&#112;&#112;&#101;&#46;&#77;&#111;&#110;&#99;&#117;&#113;&#117;&#101;&#116;&#64;&#99;&#115;&#105;&#114;&#111;&#46;&#97;&#117;</a>  </p>
<p><em>Contributor(s):</em><br />
Zhiliang Chen <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#122;&#104;&#105;&#108;&#105;&#97;&#110;&#103;&#46;&#99;&#104;&#101;&#110;&#64;&#117;&#110;&#115;&#119;&#46;&#101;&#100;&#117;&#46;&#97;&#117;">&#122;&#104;&#105;&#108;&#105;&#97;&#110;&#103;&#46;&#99;&#104;&#101;&#110;&#64;&#117;&#110;&#115;&#119;&#46;&#101;&#100;&#117;&#46;&#97;&#117;</a><br />
Sonika Tyagi <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#83;&#111;&#110;&#105;&#107;&#97;&#46;&#84;&#121;&#97;&#103;&#105;&#64;&#97;&#103;&#114;&#102;&#46;&#111;&#114;&#103;&#46;&#97;&#117;">&#83;&#111;&#110;&#105;&#107;&#97;&#46;&#84;&#121;&#97;&#103;&#105;&#64;&#97;&#103;&#114;&#102;&#46;&#111;&#114;&#103;&#46;&#97;&#117;</a>  </p>
<hr />
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<p>In this tutorial we will take raw sequencing reads and <em>de novo</em>
assemble them into contigs. We will also explore the internal assembly
graph structure to aid our understanding of why assemblies are
incomplete.</p>
<hr />
<h2 id="before-the-assembly">Before the assembly<a class="headerlink" href="#before-the-assembly" title="Permanent link">&para;</a></h2>
<h3 id="sequence-data">Sequence data<a class="headerlink" href="#sequence-data" title="Permanent link">&para;</a></h3>
<p>We have sequenced the genome of a bacterium using paired-end chemistry
on an Illumina HiSeq 2000 instrument at the University of Oxford and
publicly available as SRR2054105 the Europen Nucleotide Archive (ENA).</p>
<p>Let’s have a look at the datasets by doing:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span></pre></div></td><td class="code"><div><pre><span></span><code> cd /home/trainee/bacterial
 ls
</code></pre></div></td></tr></table></div>

<p>You should see the following two files:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span></pre></div></td><td class="code"><div><pre><span></span><code>R1.fastq.gz
R2.fastq.gz
</code></pre></div></td></tr></table></div>

<div class="admonition note">
<p class="admonition-title">Questions</p>
<p>How many reads are in this data set?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>2100778 pairs</p>
</details>
</div>
<p>What is the yield in basepairs?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>210077800 bp = 210 Mbp</p>
</details>
</div>
<p>Assuming an average bacterial genome size of 4 Mbp, what depth of coverage do we have?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>210 / 4 = 53x</p>
</details>
</div>
</div>
<h3 id="trim-or-clip-reads">Trim or clip reads<a class="headerlink" href="#trim-or-clip-reads" title="Permanent link">&para;</a></h3>
<p>There are two distinct reasons one may wish to trim or clip the raw
reads.</p>
<ol>
<li>
<p>Low quality bases typically occur toward the end of Illumina reads.
    The lower the quality score, the higher the chance that the base is
    incorrect. This introduces false k-mers into the assembly process. A
    good assembler should handle these gracefully.</p>
</li>
<li>
<p>Sequencing adaptors are artificial sequence that can occur at the
    end of reads that came from fragments of DNA that were shorter than
    desired. They were not in the original genome and their presence in
    lots of reads will totally confuse the assembler.</p>
</li>
</ol>
<p>Normally one would use <code>fastqc</code> or similar to determine whether quality
trimming is warranted and for the presence of sequencing adaptors. The
data set in this exercise is 99% free of adaptors and is good quality
overall so we will skip those steps.</p>
<div class="admonition note">
<p class="admonition-title">Questions</p>
<p>How do you know which adaptor sequence to trim?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>You either ask the sequencing provider or use a tool that tries all known adapters.</p>
</details>
</div>
<p>How are are quality values in <code>FASTQ</code> files calculated? Can we trust them?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>The are calculated using formulas calibrated to the physical measurements the sequencing instrument takes. In the case of Illumina sequencing, these measurements will include pixel light intensities from the images the camera/scanner takes of the flow cell.</p>
</details>
</div>
</div>
<h3 id="allocate-yourself-to-a-k-value-on-the-spreadsheet">Allocate yourself to a <code>K</code> value on the spreadsheet<a class="headerlink" href="#allocate-yourself-to-a-k-value-on-the-spreadsheet" title="Permanent link">&para;</a></h3>
<p>An important parameter to most genome assemblers is <code>K</code> which is the
k-mer size used to construct the <em>de Bruijn</em> graph (pronounced <em>de
Brown</em>).</p>
<p>Please go to this shared online <a href="https://docs.google.com/spreadsheets/d/1iFbCCihawpY1LClsAB-OJ66lyeW7EsdJyaMo1HCetF8/edit?usp=sharing">spreadsheet</a>
and choose <strong>a K value</strong> and put your name next to it.</p>
<hr />
<h2 id="assemble-the-reads-using-velvet">Assemble the reads using Velvet<a class="headerlink" href="#assemble-the-reads-using-velvet" title="Permanent link">&para;</a></h2>
<p>The <code>Velvet</code> software performs the assembly in two steps. The first
<code>velveth</code> step hashes the reads. The second <code>velvetg</code> step builds,
cleans and traverses the graph.</p>
<p>Choose an output folder <code>DIR</code> and use the <code>K</code> value you were allocated
on the spreadsheet and assemble the reads.</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>velveth DIR K -shortPaired -fastq.gz -separate R1.fastq.gz R2.fastq.gz
</code></pre></div></td></tr></table></div>

<p>Description of the arguments used in the command:</p>
<blockquote>
<p><strong>DIR</strong>: Directory name you chose to write the output to<br />
<strong>K</strong>: K-mer value; it must be an odd number<br />
<strong>-shortPaired</strong>: The read types are short and paired-end<br />
<strong>-fastq.gz</strong>: The format of the input files<br />
<strong>-separate</strong>: R1 and R2 reads are in separate files<br />
<strong>input file names</strong>: input FASTQ file names  </p>
</blockquote>
<p>Run the velvetg step now:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>/usr/bin/time -f &quot;%e&quot; velvetg DIR -exp_cov auto -cov_cutoff auto
</code></pre></div></td></tr></table></div>

<p>Description of the arguments used in the command:</p>
<blockquote>
<p><strong>time</strong>: to capture the <code>velvetg</code> command run time<br />
<strong>DIR</strong>: Directory name you chose to write the velvet output to<br />
<strong>-exp_cov</strong>: expected coverage is set to <code>auto</code>. To be determined by <code>velvet</code><br />
<strong>-cov_cutoff</strong>: coverage cutoff is set to <code>auto</code>. To be determined by <code>velvet</code>  </p>
</blockquote>
<h3 id="add-your-results-to-the-spreadsheet">Add your results to the spreadsheet<a class="headerlink" href="#add-your-results-to-the-spreadsheet" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th><strong>Column</strong></th>
<th><strong>Where to find the value</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>K-mer size</td>
<td>You chose this earlier and used it in the velveth command</td>
</tr>
<tr>
<td>How long it took to run</td>
<td>Look at the final output line for the user time in seconds: NN.N</td>
</tr>
<tr>
<td>Average K-mer coverage</td>
<td>Look for Estimated Coverage = NN.N</td>
</tr>
<tr>
<td>Number of contigs</td>
<td>Look for text Final graph has NNN nodes</td>
</tr>
<tr>
<td>N50 contig size</td>
<td>Look for n50 of NNNNN</td>
</tr>
<tr>
<td>Largest contig size</td>
<td>Look for max NNNNN</td>
</tr>
<tr>
<td>Total number of basepairs (sum of contig lengths)</td>
<td>Look for total NNNNNNN</td>
</tr>
</tbody>
</table>
<h3 id="effect-of-k-on-assembly">Effect of K on assembly<a class="headerlink" href="#effect-of-k-on-assembly" title="Permanent link">&para;</a></h3>
<p>Examine the spreadsheet and look for patterns in the tabular data and
corresponding bar/line charts.</p>
<p>How does K affect each of the statistics? Which value of K do you think
is doing the best job? Why?</p>
<h3 id="output-files">Output Files<a class="headerlink" href="#output-files" title="Permanent link">&para;</a></h3>
<p><strong>Table:</strong> Output Files</p>
<table>
<thead>
<tr>
<th><strong>Filename</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>contigs.fa</td>
<td>This is a FASTA file with your contigs</td>
</tr>
<tr>
<td>Log</td>
<td>Has most of the metrics in it that we recorded</td>
</tr>
<tr>
<td>stats.txt</td>
<td>TSV file of length and coverage of individual contigs</td>
</tr>
<tr>
<td>Sequences</td>
<td>A copy of the input FASTQ sequences in FASTA format</td>
</tr>
<tr>
<td>Pregraph Roadmaps Graph2</td>
<td>Interim assembly graph structure</td>
</tr>
<tr>
<td>LastGraph</td>
<td>Final graph structure</td>
</tr>
</tbody>
</table>
<p>Let’s examine the <code>stats.txt</code> file and look at the <code>short1_cov</code> column
which is the k-mer coverage of each contig:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>cut -f6 dir/stats.txt | less
</code></pre></div></td></tr></table></div>

<p>Press <code>SPACE</code> and <code>b</code> to scroll up and down, and press <code>q</code> to quote.</p>
<div class="admonition note">
<p class="admonition-title">Questions</p>
<p>What do you notice about the k-mer coverages?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>They all appear to have a similar value, but there are some that do not fit the pattern.</p>
</details>
</div>
<p>What do the outliers correspond to?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>Repetitive elements of the genome including gene duplications. Also
 replicons with differing copy numbers, like bacterial plasmids.</p>
</details>
</div>
</div>
<div class="highlight"><pre><span></span><code>grep<span class="w"> </span>NN<span class="w"> </span>dir/contigs.fa
</code></pre></div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why is there N letters in the assembly?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>Paired-end reads come from each end of the same original fragment of DNA.
The unsequenced gap in the middle is variable, and unknown.
The assembler sometimes struggles to resolve these gaps.
So it knows two sections of the genome are connected, but it doesn’t
quite know what is between them. So it pads them with its best guess
of how many bases there are, and uses N to denote them as unknown.</p>
</details>
</div>
</div>
<hr />
<h2 id="visualising-the-assembly-graph-using-bandage">Visualising the assembly graph using Bandage<a class="headerlink" href="#visualising-the-assembly-graph-using-bandage" title="Permanent link">&para;</a></h2>
<p>The final graph that <code>Velvet</code> uses is stored in the file <code>LastGraph</code>.
The <code>Bandage</code> software allows us to view and explore the assembly graph.</p>
<ol>
<li>
<p>Start <code>Bandage</code>.</p>
</li>
<li>
<p>Go to <code>File -&gt; Load Graph</code> and load the <code>LastGraph</code> from your
    assembly in <code>DIR</code>. This may take a little while so be patient.</p>
</li>
<li>
<p>Maximize your window to fill up the whole screen.</p>
</li>
<li>
<p>Click <code>Draw graph</code> on the left hand panel.</p>
</li>
<li>
<p>Change <code>Random colours</code> to <code>Colour by read depth</code> on the left hand
    panel.</p>
</li>
</ol>
<p>Now get up out of your chair and walks around and look at all the
different graphs the other participants obtained with different values
of <code>K</code>.</p>
<div class="admonition note">
<p class="admonition-title">Questions</p>
<p>How does <code>K</code> affect the graph?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>Small K produces higher k-mer coverage but also more connectivity as smaller K is more ambiguous.</p>
</details>
</div>
<p>What would the graph look like in an ideal situtation?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>If the genome had M replicons, we would like to see only M sub-graphs. Each sub-graph would be linear or circular, depending on the form of the replicon in the original organism.</p>
</details>
</div>
<p>Why didn’t anyone achieve perfection?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>Short reads are unable to disambiguate repeats longer than the read
length (or read span). Most genomes have repeats beyond 500 bp.</p>
</details>
</div>
</div>
<p>Here is another <a href="https://github.com/rrwick/Bandage/wiki/Effect-of-kmer-size">example</a>
from the Bandage web site on how K can affect assembly graphs.</p>
<h3 id="explore-the-graph">Explore the graph<a class="headerlink" href="#explore-the-graph" title="Permanent link">&para;</a></h3>
<p><code>Bandage</code> is designed to be an interactive tool. It allows you to see
relationships between parts of your genome that are lost when you only
look at the FASTA file of contigs.</p>
<ul>
<li>
<p>Zoom in using the <code>Zoom</code> up/down button in the left hand panel</p>
</li>
<li>
<p>Pan around by holding down the Option/Windows key and dragging on
    the background</p>
</li>
<li>
<p>Move nodes out of the way by selecting and dragging</p>
</li>
</ul>
<p>Feel free to play around a bit and ask questions.</p>
<h3 id="features-of-the-assembly-graph">Features of the assembly graph<a class="headerlink" href="#features-of-the-assembly-graph" title="Permanent link">&para;</a></h3>
<p>The graph is quite tangled. The long stretches correspond to contigs.
The intersections correspond to shared k-mers in the graph, which occur
due to repeated sequences within the genome.</p>
<p>Select an node (rectangle) in the graph. It’s length is reported in the
right hand panel as<br />
<code>Length: NNN</code>.</p>
<p>On the menu choose <code>Output -&gt; Web BLAST selected node</code>. Your browser
should open the <a href="http://blast.ncbi.nlm.nih.gov/Blast.cgi?PROGRAM=blastn&amp;PAGE_TYPE=BlastSearch&amp;LINK_LOC=blasthome">NCBI BLAST Website</a>.</p>
<p>Click the <code>BLAST</code> button at the bottom, and wait for the result.</p>
<div class="admonition note">
<p class="admonition-title">Questions</p>
<p>Did your node match anything in the Genbank database?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>You will need to examine the BLAST report on the trainee browser window.</p>
</details>
</div>
<p>Can you determine what species of bacteria was sequenced?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>The top hit of the small segment of DNA the trainee chose may not
necessarily reflect the true species of bacteria. But if multiple
segments produce consistent top hits to the same species you would have some confidence.</p>
</details>
</div>
</div>
<hr />
<h2 id="conclusion">Conclusion<a class="headerlink" href="#conclusion" title="Permanent link">&para;</a></h2>
<p>You should now:</p>
<ul>
<li>
<p>know how to use Velvet to assemble a simple genome from Illumina
    sequences</p>
</li>
<li>
<p>understand the role of the k-mer length K in the assembly process</p>
</li>
<li>
<p>be able to relate the graph structure to the final contigs</p>
</li>
<li>
<p>realize the limitations of short read sequences with respect to
    genome complexity</p>
</li>
</ul>
<p>Thank you.</p>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      &copy; The authors; 2017-
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../../..", "features": [], "search": "../../../assets/javascripts/workers/search.12658920.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.5cf534bf.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
  </body>
</html>