
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
        <meta name="author" content="Authors">
      
      
      
        <link rel="prev" href="../../denovo-module-bac/bac_asm/">
      
      
        <link rel="next" href="../../denovo-module-scaffolding/scaffold/">
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.2">
    
    
      
        <title>Eukaryote Assembly - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.f56500e0.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#key-learning-outcomes" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-header__button md-logo" aria-label="Markdown Manuals for Bioinformatics Training Platform (BTP)" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Markdown Manuals for Bioinformatics Training Platform (BTP)
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Eukaryote Assembly
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-nav__button md-logo" aria-label="Markdown Manuals for Bioinformatics Training Platform (BTP)" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../trainers/trainers/" class="md-nav__link">
        The Trainers
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          Timetable
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Timetable" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Timetable
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_introNGS/" class="md-nav__link">
        Intro to NGS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_cancer/" class="md-nav__link">
        Cancer Genomics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_longRead/" class="md-nav__link">
        Long Read
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../preamble/" class="md-nav__link">
        Workshop Information
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Introduction to NGS Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Introduction to NGS Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Introduction to NGS Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-cli/commandline/" class="md-nav__link">
        Introduction to Command Line
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-qc/ngs-qc/" class="md-nav__link">
        Data Quality
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-mapping/alignment/" class="md-nav__link">
        Read Alignment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-chip-seq/chip-seq/" class="md-nav__link">
        ChIP-seq
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-rna-seq/rna-seq/" class="md-nav__link">
        RNA-Seq
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-bio/biol/" class="md-nav__link">
        NGS Biological Insight
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-velvet/velvet/" class="md-nav__link">
        de novo Genome Assembly
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-denovo-canu/denovo_canu/" class="md-nav__link">
        DeNovo Canu
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" >
      
      
      
      
        <label class="md-nav__link" for="__nav_6">
          Cancer Genomics Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Cancer Genomics Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Cancer Genomics Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-alignment/alignment/" class="md-nav__link">
        Read Alignment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-snv/snv/" class="md-nav__link">
        Single Nucleotide Variant Calling and Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-cnv/cnv-tut/" class="md-nav__link">
        Copy Number Variation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-sv/sv_tut/" class="md-nav__link">
        Structural Variant Analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-viz/visu/" class="md-nav__link">
        Variant Visualisation
      </a>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6_6" type="checkbox" id="__nav_6_6" >
      
      
      
      
        <label class="md-nav__link" for="__nav_6_6">
          Somatic Mutations
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Somatic Mutations" data-md-level="2">
        <label class="md-nav__title" for="__nav_6_6">
          <span class="md-nav__icon md-icon"></span>
          Somatic Mutations
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-somatic/01_signatures/" class="md-nav__link">
        Assessing somatic mutational signatures
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-somatic/02_intogen/" class="md-nav__link">
        Cohort analysis for the identification of driver genes
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_7" type="checkbox" id="__nav_7" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_7">
          DeNovo Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="DeNovo Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          DeNovo Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../denovo-module-bac/bac_asm/" class="md-nav__link">
        Bacterial Assembly
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Eukaryote Assembly
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Eukaryote Assembly
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav" aria-label="Resources You’ll be Using">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#author-information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#first-pass-genome-assembly" class="md-nav__link">
    First Pass Genome Assembly
  </a>
  
    <nav class="md-nav" aria-label="First Pass Genome Assembly">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fusarium-first-pass-with-a-goal" class="md-nav__link">
    Fusarium first pass with a goal
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#task-11-first-pass-assembly-k71" class="md-nav__link">
    Task 1.1: First pass assembly, k=71
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#task-12-first-pass-assembly-k27" class="md-nav__link">
    Task 1.2: First pass assembly, k=27
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    References
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../denovo-module-scaffolding/scaffold/" class="md-nav__link">
        Scaffolding
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_8" type="checkbox" id="__nav_8" >
      
      
      
      
        <label class="md-nav__link" for="__nav_8">
          Metagenomics Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Metagenomics Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_8">
          <span class="md-nav__icon md-icon"></span>
          Metagenomics Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-qc/ngs-qc/" class="md-nav__link">
        Metagenomics QC
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-tax/tax/" class="md-nav__link">
        Metagenomics Taxonomic
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-vis/vis/" class="md-nav__link">
        Metagenomics Visualisation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-wga/wga/" class="md-nav__link">
        Metagenomics WGA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-fda/fda/" class="md-nav__link">
        Metagenomics Functional Analysis
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav" aria-label="Resources You’ll be Using">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#author-information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#first-pass-genome-assembly" class="md-nav__link">
    First Pass Genome Assembly
  </a>
  
    <nav class="md-nav" aria-label="First Pass Genome Assembly">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fusarium-first-pass-with-a-goal" class="md-nav__link">
    Fusarium first pass with a goal
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#task-11-first-pass-assembly-k71" class="md-nav__link">
    Task 1.1: First pass assembly, k=71
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#task-12-first-pass-assembly-k27" class="md-nav__link">
    Task 1.2: First pass assembly, k=27
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    References
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



  <h1>Eukaryote Assembly</h1>

<h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<p>After completing this module the trainee should be able to:</p>
<ul>
<li>
<p>Assess general assembly approach, kmer spectra and biases.</p>
</li>
<li>
<p>Visually inspect the kmer spectra and KAT plots</p>
</li>
<li>
<p>Run a first pass eukaryotic assembly and do goal checks</p>
</li>
<li>
<p>Develop validation metrics or tools for NGS data and assembly.</p>
</li>
<li>
<p>Improving methods and pipelines for genome assembly.</p>
</li>
<li>
<p>Convince the lab guys to tweak protocols.</p>
</li>
</ul>
<hr />
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>Kmer Analysis Tool kit:<br />
<a href="https://github.com/TGAC/KAT">https://github.com/TGAC/KAT</a></p>
<p>Nextclip:<br />
<a href="https://github.com/richardmleggett/nextclip">https://github.com/richardmleggett/nextclip</a></p>
<p>Abyss:<br />
<a href="http://www.bcgsc.ca/platform/bioinfo/software/abyss">http://www.bcgsc.ca/platform/bioinfo/software/abyss</a></p>
<p>Soap Denovo:<br />
<a href="http://soap.genomics.org.cn/soapdenovo.html">http://soap.genomics.org.cn/soapdenovo.html</a></p>
<p>SOAPec:<br />
<a href="http://soap.genomics.org.cn/about.html">http://soap.genomics.org.cn/about.html</a></p>
<p>BLAST:<br />
<a href="http://blast.ncbi.nlm.nih.gov/Blast.cgi">http://blast.ncbi.nlm.nih.gov/Blast.cgi</a></p>
<hr />
<h2 id="author-information">Author Information<a class="headerlink" href="#author-information" title="Permanent link">&para;</a></h2>
<p><em>Primary Author(s):</em><br />
Bernardo Clavijo, TGAC <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#66;&#101;&#114;&#110;&#97;&#114;&#100;&#111;&#46;&#67;&#108;&#97;&#118;&#105;&#106;&#111;&#64;&#116;&#103;&#97;&#99;&#46;&#97;&#99;&#46;&#117;&#107;">&#66;&#101;&#114;&#110;&#97;&#114;&#100;&#111;&#46;&#67;&#108;&#97;&#118;&#105;&#106;&#111;&#64;&#116;&#103;&#97;&#99;&#46;&#97;&#99;&#46;&#117;&#107;</a></p>
<p><em>Contributor(s):</em><br />
Sonika Tyagi, AGRF <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#115;&#111;&#110;&#105;&#107;&#97;&#46;&#116;&#121;&#97;&#103;&#105;&#64;&#97;&#103;&#114;&#102;&#46;&#111;&#114;&#103;&#46;&#97;&#117;">&#115;&#111;&#110;&#105;&#107;&#97;&#46;&#116;&#121;&#97;&#103;&#105;&#64;&#97;&#103;&#114;&#102;&#46;&#111;&#114;&#103;&#46;&#97;&#117;</a><br />
Zhiliang Chan <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#122;&#104;&#105;&#108;&#105;&#97;&#110;&#103;&#64;&#117;&#110;&#115;&#119;&#46;&#101;&#100;&#117;&#46;&#97;&#117;">&#122;&#104;&#105;&#108;&#105;&#97;&#110;&#103;&#64;&#117;&#110;&#115;&#119;&#46;&#101;&#100;&#117;&#46;&#97;&#117;</a></p>
<hr />
<h2 id="first-pass-genome-assembly">First Pass Genome Assembly<a class="headerlink" href="#first-pass-genome-assembly" title="Permanent link">&para;</a></h2>
<p>Assuming by now you are familiar with the general concept of <em>de novo</em>
assembly, kmers and the de Bruijn graph based assembler. In this
tutorial we will use ABySS to perform the first pass assembly of a
eukaryotic genome and look at various parameters to assess the
information content of the input data and choice of assembly parameters.
sequence data.  </p>
<p>Genome assembly is a challenging problem requiring heavy computational
resources, expertise and time. Before you begin the process of <em>de novo</em>
assembly there are a number of points you need to consider:</p>
<ul>
<li>
<p>What is the objective of your assembly experiment? What biological
question(s) you have?  </p>
</li>
<li>
<p>Is assembly strictly necessary for the purpose in question?  </p>
</li>
<li>
<p>Do you have right kind of data and enough coverage to start with?</p>
</li>
<li>
<p>Do you have suitable computational resources to run this assembly?</p>
</li>
</ul>
<p>Remember that the assembly is just a probabilistic model of a genome,
condensing the information from the experimental evidence. All the
information is already present in the experimental results. The goal of
the assembly is to find the right motifs, the correct number of times,
in correct order and position.</p>
<h3 id="fusarium-first-pass-with-a-goal"><em>Fusarium</em> first pass with a goal<a class="headerlink" href="#fusarium-first-pass-with-a-goal" title="Permanent link">&para;</a></h3>
<p>Goal: Identify a fusarium sample is &ldquo;closer&rdquo; to <em>F. graminearum</em> or
<em>F. pseudograminearum</em></p>
<ul>
<li>
<p>Previous knowledge</p>
<ul>
<li><em>F. graminearum</em> has a cluster producing PKS6 and NRSP7, while <em>F. pseudograminearum</em> produces PKS40 and NRPS32</li>
</ul>
</li>
<li>
<p>Data</p>
<ul>
<li>
<p>Proteins sequences for:</p>
<ol>
<li>
<p><em>F. graminearum</em> (non necrotrophic): PKS6 and NRSP7</p>
</li>
<li>
<p><em>F. pseudograminearum</em> (necrotrophic): PKS40 and NRPS32</p>
</li>
</ol>
</li>
</ul>
</li>
<li>
<p>Strategy</p>
<ul>
<li>Check PKS6-NRSP7 and PKS40-NRPS32 cluster presence.</li>
</ul>
</li>
<li>
<p>Assembly goals:</p>
<ul>
<li>
<p>To capture a good enough representation of the protein-coding space to get blast matches.</p>
</li>
<li>
<p>To accurately represent the relevant whole cluster loci in a single sequence.</p>
</li>
</ul>
</li>
</ul>
<hr />
<h2 id="prepare-the-environment">Prepare the Environment<a class="headerlink" href="#prepare-the-environment" title="Permanent link">&para;</a></h2>
<p>Open the Terminal. First, go to the right folder, where the data are
stored.</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span></pre></div></td><td class="code"><div><pre><span></span><code>cd /home/trainee/eukaryotic
ls
</code></pre></div></td></tr></table></div>

<h2 id="task-11-first-pass-assembly-k71">Task 1.1: First pass assembly, k=71<a class="headerlink" href="#task-11-first-pass-assembly-k71" title="Permanent link">&para;</a></h2>
<p>Let’s assemble <em>Fusarium</em> with abyss, k=71</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span>
<span class="normal">3</span>
<span class="normal">4</span></pre></div></td><td class="code"><div><pre><span></span><code>cd /home/trainee/eukaryotic
mkdir abyss_k71
cd abyss_k71
abyss-pe in=&quot;../CS3270_A8733_GCCAAT_L001_R1.fastq ../CS3270_A8733_GCCAAT_L001_R2.fastq&quot; k=71 name=CS3270_abyss_k71 np=4 &gt; CS3270_abyss_k71.log 2&gt;&amp;1
</code></pre></div></td></tr></table></div>

<p>Description of the arguments used in the command:</p>
<blockquote>
<p><strong>k</strong>: kmer size<br />
<strong>np</strong>: number of processors to be used<br />
<strong>sequence file names</strong>: R1 and R2 reads of a paired end sequence data  </p>
</blockquote>
<p><br>
Let’s look at the statistics of the assembly we just did.</p>
<p>Ok, there are no stats available in the folder, but we can always use
<code>abyss-fac</code> to get the stats:</p>
<div class="highlight"><pre><span></span><code>abyss-fac CS3270_abyss_k71-*tigs.fa  | tee CS3270_abyss_k71-stats.tab
less CS3270_abyss_k71-stats.tab
</code></pre></div>
<p><br>
<strong>Table 1:</strong> Statistics of fusarium assembly by ABySS using k=71</p>
<table>
<thead>
<tr>
<th><strong>n</strong></th>
<th><strong>n:500</strong></th>
<th><strong>L50</strong></th>
<th><strong>min</strong></th>
<th><strong>N80</strong></th>
<th><strong>N50</strong></th>
<th><strong>N20</strong></th>
<th><strong>E-size</strong></th>
<th><strong>max</strong></th>
<th><strong>sum</strong></th>
<th><strong>name</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>27</td>
<td>13</td>
<td>2</td>
<td>970</td>
<td>6004</td>
<td>13202</td>
<td>52602</td>
<td>28712</td>
<td>52602</td>
<td>112849</td>
<td>CS3270_abyss_k71-unitigs.fa</td>
</tr>
<tr>
<td>5</td>
<td>1</td>
<td>1</td>
<td>128429</td>
<td>128429</td>
<td>128429</td>
<td>128429</td>
<td>128429</td>
<td>128429</td>
<td>128429</td>
<td>CS3270_abyss_k71-contigs.fa</td>
</tr>
</tbody>
</table>
<p><br></p>
<div class="admonition note">
<p class="admonition-title">Questions</p>
<p>How many unitigs/contigs do you have in the assembly?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>27/5</p>
</details>
</div>
<p>What are the length statistics of your assembly?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>In the table above</p>
</details>
</div>
<p>Does it match what you think before the assembly and why?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>No</p>
</details>
</div>
</div>
<p><br>
The assembly is looking strange! It’s time for some analysis:</p>
<p>Check frequencies for kmers kept/discarded/etc.</p>
<p>Check spectra-cn and compare with expectations. Let’s do this by the
following commands:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span></pre></div></td><td class="code"><div><pre><span></span><code>less CS3270_abyss_k71.log
less coverage.hist
</code></pre></div></td></tr></table></div>

<p>We will now plot the values from the <code>coverage.hist</code>:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span>
<span class="normal">3</span>
<span class="normal">4</span>
<span class="normal">5</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="nv">gnuplot</span><span class="w"> </span><span class="o">&lt;</span><span class="nv">Press</span><span class="w"> </span><span class="nv">enter</span><span class="o">&gt;</span>
<span class="nv">set</span><span class="w"> </span><span class="nv">xrange</span><span class="w"> </span>[<span class="mi">0</span>:<span class="mi">200</span>]
<span class="nv">set</span><span class="w"> </span><span class="nv">yrange</span><span class="w"> </span>[<span class="mi">0</span>:<span class="mi">5000</span>]
<span class="nv">plot</span><span class="w"> </span><span class="s2">&quot;coverage.hist&quot;</span>
<span class="k">exit</span>
</code></pre></div></td></tr></table></div>

<p>Looks like we are not assembling this bit, let’s have another look at
the spectra</p>
<div class="highlight"><pre><span></span><code>kat comp -o reads_vs_abyss_k71 -t 4 -C -D --d1_bins 2000 &#39;../*.fastq&#39; CS3270_abyss_k71-contigs.fa
</code></pre></div>
<p>Description of the arguments used in the command:</p>
<blockquote>
<p><strong>o</strong>: Path prefix for files generated by this program.<br />
<strong>t</strong>: The number of threads to use.<br />
<strong>C</strong>: Whether the jellyfish hash for input 1 contains K-mers produced for both strands<br />
<strong>D</strong>: Whether the jellyfish hash for input 2 contains K-mers produced for both strands<br />
<strong>--d1_bins</strong>: Number of bins for the first dataset. i.e. number of rows in the matrix  </p>
</blockquote>
<p><br></p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>kat plot spectra-cn -y 600 -x 2000 -o reads_vs_abyss1-main.mx.spectra-cn_2000.png reads_vs_abyss_k71-main.mx
</code></pre></div></td></tr></table></div>

<p>Description of the arguments used in the command:</p>
<blockquote>
<p><strong>x</strong>: Maximum value for the x-axis (default value auto calculated from matrix, otherwise 1000)<br />
<strong>y</strong>: Maximum value for the y-axis (default value auto calculated from matrix if possible, otherwise, 1000000)<br />
<strong>o</strong>: The path to the output file</p>
</blockquote>
<p><br>
The kmer spectra for <em>Fusarium</em> assembly with <code>abyss</code>, k=71 should be looking like this:</p>
<p><img alt="image" src="../images/abyss-k71.png" /> <strong>Figure 1</strong>: Kmer spectrum for Fusarium assembly with abyss, k=71, coloured by the
frequencies as found on the reference assembly.  </p>
<p><br>
Take the output and BLAST it in NCBI. What is it? Surprising?</p>
<p>Choosing a wrong k value (too large in this case) and just running a
typical assembly job, we can end up with something quite more
interesting. It is easy by comparison to spot some missing content,
alongside duplications and triplications (and quadruplications and so
on) that should not be there. This assembly will get us nowhere, let’s
choose a lower K to gain coverage and start again.</p>
<h2 id="task-12-first-pass-assembly-k27">Task 1.2: First pass assembly, k=27<a class="headerlink" href="#task-12-first-pass-assembly-k27" title="Permanent link">&para;</a></h2>
<p>We now assemble <em>Fusarium</em> with <code>abyss</code> and k=27:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span>
<span class="normal">3</span>
<span class="normal">4</span></pre></div></td><td class="code"><div><pre><span></span><code>cd /home/trainee/eukaryotic
mkdir abyss_k27
cd abyss_k27
abyss-pe in=&quot;../CS3270_A8733_GCCAAT_L001_R1.fastq ../CS3270_A8733_GCCAAT_L001_R2.fastq&quot; k=27 name=CS3270_abyss_k27 np=4 &gt; CS3270_abyss_k27.log 2&gt;&amp;1
</code></pre></div></td></tr></table></div>

<p>Let’s look at the stats by doing:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>less CS3270_abyss_k27-stats.tab
</code></pre></div></td></tr></table></div>

<p><br>
The stats look better:  </p>
<p><strong>Table 2:</strong> Statistics of fusarium assembly by ABySS using k=27.</p>
<table>
<thead>
<tr>
<th><strong>n</strong></th>
<th><strong>n:500</strong></th>
<th><strong>L50</strong></th>
<th><strong>min</strong></th>
<th><strong>N80</strong></th>
<th><strong>N50</strong></th>
<th><strong>N20</strong></th>
<th><strong>E-size</strong></th>
<th><strong>max</strong></th>
<th><strong>sum</strong></th>
<th><strong>name</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>30645</td>
<td>2717</td>
<td>430</td>
<td>502</td>
<td>11354</td>
<td>25336</td>
<td>47966</td>
<td>31027</td>
<td>147694</td>
<td>36.14e6</td>
<td>CS3270_abyss_k27-unitigs.fa</td>
</tr>
<tr>
<td>21511</td>
<td>350</td>
<td>33</td>
<td>527</td>
<td>157565</td>
<td>338989</td>
<td>630228</td>
<td>407098</td>
<td>1265237</td>
<td>36.52e6</td>
<td>CS3270_abyss_k27-contigs.fa</td>
</tr>
<tr>
<td>21327</td>
<td>205</td>
<td>17</td>
<td>527</td>
<td>332444</td>
<td>716132</td>
<td>1265237</td>
<td>791882</td>
<td>1880850</td>
<td>36.51e6</td>
<td>CS3270_abyss_k27-scaffolds.fa</td>
</tr>
</tbody>
</table>
<p><br>
Let’s check a bit anyway:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span></pre></div></td><td class="code"><div><pre><span></span><code>less CS3270_abyss_k27.log
less coverage.hist
</code></pre></div></td></tr></table></div>

<p>How is the coverage plot looking now?</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span>
<span class="normal">3</span>
<span class="normal">4</span>
<span class="normal">5</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="nv">gnuplot</span>
<span class="nv">gnuplot</span><span class="o">&gt;</span><span class="w"> </span><span class="nv">set</span><span class="w"> </span><span class="nv">xrange</span><span class="w"> </span>[<span class="mi">0</span>:<span class="mi">50</span>]
<span class="nv">gnuplot</span><span class="o">&gt;</span><span class="w"> </span><span class="nv">set</span><span class="w"> </span><span class="nv">xrange</span><span class="w"> </span>[<span class="mi">0</span>:<span class="mi">4000000</span>]
<span class="nv">gnuplot</span><span class="o">&gt;</span><span class="w"> </span><span class="nv">plot</span><span class="w"> </span><span class="s2">&quot;coverage.hist&quot;</span>
<span class="nv">gnuplot</span><span class="o">&gt;</span><span class="w"> </span><span class="k">exit</span>
</code></pre></div></td></tr></table></div>

<p>K-mer spectrum:</p>
<p><div class="highlight"><pre><span></span><code>kat plot spectra-cn -y 1000 -x 1000 -o reads_vs_abyss1-main.mx.spectra-cn_noabsent.png reads_vs_abyss1-main.mx
kat comp -o reads_vs_abyss_k27 -t 4 -C -D &#39;../*.fastq&#39; CS3270_abyss_k27-scaffolds.fa
</code></pre></div>
<br>
<img alt="image" src="../images/abyss-k27.png" /> <strong>Figure 2:</strong> Kmer spectrum for Fusarium assembly with abyss, k=27, coloured by the
frequencies as found on the reference assembly.<br />
<br></p>
<div class="admonition note">
<p class="admonition-title">Questions</p>
<p>Any tools you can use to check kmer spectra at any K before assembling?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>KAT</p>
</details>
</div>
<p>Can you predict what will happen if you use KAT with larger K values?</p>
</div>
<p><br>
Will the assembly answer the biological question?</p>
<p>Use BLAST or BLAT and the databases to check.</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span></pre></div></td><td class="code"><div><pre><span></span><code>makeblastdb -in CS3270_abyss_k27-scaffolds.fa -dbtype nucl
blat –t=dnax –q=prot –minIdentity=90 CS3270_abyss_k27-scaffolds.fa test_genes.fasta out.psl
</code></pre></div></td></tr></table></div>

<hr />
<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<ol>
<li>De novo genome assembly: what every biologist should know Nature
    Methods 9, 333 – 337 (2012) doi:10.1038/nmeth.1935</li>
</ol>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      &copy; The authors; 2017-
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../../..", "features": [], "search": "../../../assets/javascripts/workers/search.12658920.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.5cf534bf.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
  </body>
</html>