
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
        <meta name="author" content="Authors">
      
      
      
        <link rel="prev" href="../../metagenomics-module-tax/tax/">
      
      
        <link rel="next" href="../../metagenomics-module-wga/wga/">
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.2">
    
    
      
        <title>Metagenomics Visualisation - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.f56500e0.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#analysis-visualisation" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-header__button md-logo" aria-label="Markdown Manuals for Bioinformatics Training Platform (BTP)" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Markdown Manuals for Bioinformatics Training Platform (BTP)
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Metagenomics Visualisation
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-nav__button md-logo" aria-label="Markdown Manuals for Bioinformatics Training Platform (BTP)" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../trainers/trainers/" class="md-nav__link">
        The Trainers
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          Timetable
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Timetable" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Timetable
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_introNGS/" class="md-nav__link">
        Intro to NGS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_cancer/" class="md-nav__link">
        Cancer Genomics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_longRead/" class="md-nav__link">
        Long Read
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../preamble/" class="md-nav__link">
        Workshop Information
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Introduction to NGS Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Introduction to NGS Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Introduction to NGS Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-cli/commandline/" class="md-nav__link">
        Introduction to Command Line
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-qc/ngs-qc/" class="md-nav__link">
        Data Quality
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-mapping/alignment/" class="md-nav__link">
        Read Alignment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-chip-seq/chip-seq/" class="md-nav__link">
        ChIP-seq
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-rna-seq/rna-seq/" class="md-nav__link">
        RNA-Seq
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-bio/biol/" class="md-nav__link">
        NGS Biological Insight
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-velvet/velvet/" class="md-nav__link">
        de novo Genome Assembly
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-denovo-canu/denovo_canu/" class="md-nav__link">
        DeNovo Canu
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" >
      
      
      
      
        <label class="md-nav__link" for="__nav_6">
          Cancer Genomics Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Cancer Genomics Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Cancer Genomics Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-alignment/alignment/" class="md-nav__link">
        Read Alignment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-snv/snv/" class="md-nav__link">
        Single Nucleotide Variant Calling and Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-cnv/cnv-tut/" class="md-nav__link">
        Copy Number Variation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-sv/sv_tut/" class="md-nav__link">
        Structural Variant Analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-viz/visu/" class="md-nav__link">
        Variant Visualisation
      </a>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6_6" type="checkbox" id="__nav_6_6" >
      
      
      
      
        <label class="md-nav__link" for="__nav_6_6">
          Somatic Mutations
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Somatic Mutations" data-md-level="2">
        <label class="md-nav__title" for="__nav_6_6">
          <span class="md-nav__icon md-icon"></span>
          Somatic Mutations
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-somatic/01_signatures/" class="md-nav__link">
        Assessing somatic mutational signatures
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-somatic/02_intogen/" class="md-nav__link">
        Cohort analysis for the identification of driver genes
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_7" type="checkbox" id="__nav_7" >
      
      
      
      
        <label class="md-nav__link" for="__nav_7">
          DeNovo Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="DeNovo Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          DeNovo Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../denovo-module-bac/bac_asm/" class="md-nav__link">
        Bacterial Assembly
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../denovo-module-euk/eukaryotic_assem/" class="md-nav__link">
        Eukaryote Assembly
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../denovo-module-scaffolding/scaffold/" class="md-nav__link">
        Scaffolding
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_8" type="checkbox" id="__nav_8" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_8">
          Metagenomics Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Metagenomics Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_8">
          <span class="md-nav__icon md-icon"></span>
          Metagenomics Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-qc/ngs-qc/" class="md-nav__link">
        Metagenomics QC
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-tax/tax/" class="md-nav__link">
        Metagenomics Taxonomic
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Metagenomics Visualisation
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Metagenomics Visualisation
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav" aria-label="Resources You’ll be Using">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#data-sets" class="md-nav__link">
    Data sets
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#beta-diversity-computation-and-plots" class="md-nav__link">
    Beta Diversity computation and plots
  </a>
  
    <nav class="md-nav" aria-label="Beta Diversity computation and plots">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#prepare-the-environment" class="md-nav__link">
    Prepare the environment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#emperor" class="md-nav__link">
    Emperor
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-wga/wga/" class="md-nav__link">
        Metagenomics WGA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-fda/fda/" class="md-nav__link">
        Metagenomics Functional Analysis
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav" aria-label="Resources You’ll be Using">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#data-sets" class="md-nav__link">
    Data sets
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#beta-diversity-computation-and-plots" class="md-nav__link">
    Beta Diversity computation and plots
  </a>
  
    <nav class="md-nav" aria-label="Beta Diversity computation and plots">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#prepare-the-environment" class="md-nav__link">
    Prepare the environment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#emperor" class="md-nav__link">
    Emperor
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<h1 id="analysis-visualisation">Analysis Visualisation<a class="headerlink" href="#analysis-visualisation" title="Permanent link">&para;</a></h1>
<h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<hr />
<p>After completing this module the trainee should be able to:</p>
<ul>
<li>
<p>Visualise between sample comparisons, using Emperor PCA</p>
</li>
<li>
<p>Understand the difference between weighted and unweighted analysis</p>
</li>
</ul>
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<hr />
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>Emperor: <br />
<a href="http://qiime.org/emperor/tutorial_index.html">http://qiime.org/emperor/tutorial_index.html</a></p>
<h3 id="data-sets">Data sets<a class="headerlink" href="#data-sets" title="Permanent link">&para;</a></h3>
<hr />
<p>Sutton et al. (2013). Impact of Long-Term Diesel Contamination on Soil Microbial Community Structure: <br />
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3553749/pdf/zam619.pdf">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3553749/pdf/zam619.pdf</a></p>
<p>Fierer et al. (2010). Forensic identification using skin bacterial communities: <br />
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2852011/">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2852011/</a></p>
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<hr />
<p>Good data quality and sample metadata is important for visualising
metagenomics analysis.</p>
<p>For this tutorial we are using Emperor a browser enabled scatter plot
visualisation tool. We will be using the Sutton and Fierer data sets for
viewing the principal components analysis (PCoA) from the QIIME Beta
diversity analysis (16S).</p>
<h2 id="beta-diversity-computation-and-plots">Beta Diversity computation and plots<a class="headerlink" href="#beta-diversity-computation-and-plots" title="Permanent link">&para;</a></h2>
<hr />
<p>Beta diversity represents between sample comparisons based on their
composition. The output of these comparisons is a square matrix where a
“distance” or dissimilarity is calculated between every pair of
community samples, reflecting the dissimilarity between those samples.
The data distance matrix can be then visualized with analyses such as
PCoA and hierarchical clustering. PCoA is a technique that maps the
samples in the distance matrix to a set of axes that show the maximum
amount of variation explained. The principal coordinates can be plotted
in two or three dimensions to provide an intuitive visualization of the
data structure to look at differences between the samples, and look for
similarities by sample conditions.</p>
<p>The Beta diversity workflow you ran earlier contained a number of steps:</p>
<ul>
<li>
<p>Rarifying the OTU table to remove sample heterogeneity. A rarified
    OTU table should be used so that artificial diversity induced due to
    different sampling effort is removed.</p>
</li>
<li>
<p>Calculating the Beta diversity, the unifrac weighted and unweighted
    generated principal coorodinates are created here by default.</p>
</li>
</ul>
<p>The following are the required options and inputs for Beta diversity
analysis:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span>
<span class="normal">3</span>
<span class="normal">4</span>
<span class="normal">5</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="mf">1.</span><span class="w">  </span><span class="n">Option</span><span class="w"> </span><span class="o">-</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">OTU</span><span class="w"> </span><span class="nb">tab</span><span class="n">le</span><span class="w"> </span><span class="p">(</span><span class="o">*</span><span class="mf">.</span><span class="n">biom</span><span class="p">)</span>
<span class="mf">2.</span><span class="w">  </span><span class="n">Option</span><span class="w"> </span><span class="err">–</span><span class="n">t</span><span class="p">,</span><span class="w"> </span><span class="n">phylogeny</span><span class="w"> </span><span class="n">tree</span><span class="w"> </span><span class="p">(</span><span class="o">*</span><span class="mf">.</span><span class="n">tre</span><span class="p">)</span><span class="w"> </span><span class="n">from</span><span class="w"> </span><span class="n">pick_de_novo_otus</span><span class="mf">.</span><span class="n">py</span>
<span class="mf">3.</span><span class="w">  </span><span class="n">Option</span><span class="w"> </span><span class="err">–</span><span class="n">m</span><span class="p">,</span><span class="w"> </span><span class="n">the</span><span class="w"> </span><span class="n">user</span><span class="o">-</span><span class="kd">def</span><span class="n">ined</span><span class="w"> </span><span class="n">mapping</span><span class="w"> </span><span class="n">file</span>
<span class="mf">4.</span><span class="w">  </span><span class="n">Option</span><span class="w"> </span><span class="err">–</span><span class="n">o</span><span class="p">,</span><span class="w"> </span><span class="n">the</span><span class="w"> </span><span class="n">output</span><span class="w"> </span><span class="n">directory</span>
<span class="mf">5.</span><span class="w">  </span><span class="n">Option</span><span class="w"> </span><span class="err">–</span><span class="n">e</span><span class="p">,</span><span class="w"> </span><span class="n">the</span><span class="w"> </span><span class="n">number</span><span class="w"> </span><span class="n">of</span><span class="w"> </span><span class="n">sequences</span><span class="w"> </span><span class="n">per</span><span class="w"> </span><span class="n">sample</span><span class="w"> </span><span class="p">(</span><span class="n">sequencing</span><span class="w"> </span><span class="n">depth</span><span class="p">)</span>
</code></pre></div></td></tr></table></div>

<p>You do not need to run the following command, it may overwrite the
pre-computed analysis. This is the command you ran on the Sutton sub
dataset.</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>beta_diversity_through_plots.py -i otus/otu_table.biom -m mapping.txt -o wf_bdiv_even122/ -t otus/rep_set.tre -e 122
</code></pre></div></td></tr></table></div>

<h3 id="prepare-the-environment">Prepare the environment<a class="headerlink" href="#prepare-the-environment" title="Permanent link">&para;</a></h3>
<p>The data for this practical can be found in the Taxonomy directory on
your desktop. Go to the pre-computed Beta diversity analysis in the
Taxonomy directory.</p>
<p>Open the Terminal and go to where the data is stored. Investigate the
directories available.</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span></pre></div></td><td class="code"><div><pre><span></span><code>cd ~/Desktop/Taxonomy/sutton_full_denoised/
ls –lhtr
</code></pre></div></td></tr></table></div>

<p>Open the following files: wf_bdiv_even394/unweighted_unifrac_pc.txt
and wf_bdiv_even394/weighted_unifrac_pc.txt using the ‘less’
command. The option –S with ‘less’ allows you to view files with
unwrapped lines, to escape type q (for quit). To scroll left and right,
use the arrow keys.</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span></pre></div></td><td class="code"><div><pre><span></span><code>less –S wf_bdiv_even394/unweighted_unifrac_pc.txt
less –S wf_bdiv_even394/weighted_unifrac_pc.txt
</code></pre></div></td></tr></table></div>

<p>In each file every sample is listed in the first column, and the
subsequent columns contain the value for the sample against the noted
principal coordinate. At the bottom of each Principal Coordinate column,
you will find the eigen value and percent of variation explained by the
coordinate. Note the major difference between weighted and unweighted
analysis is the inclusion of OTU abundance when calculating distances
between communities. You should use weighted if the biological question
you are trying to ask takes OTU abundance of your groups into
consideration. If some samples are forming groups with weighted, then
likely the larger or smaller abundances of several OTUs are the primary
driving force in PCoA space, but when all OTUs are considered at equal
abundance, these differences are lost (unweighted).</p>
<p>Note discrete vs. continuous only has to do with coloring of the points,
and not the calculation of UniFrac distances.</p>
<h3 id="emperor">Emperor<a class="headerlink" href="#emperor" title="Permanent link">&para;</a></h3>
<p>We can now run Emperor to prepare the PCoA plots for visualisation.</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span>
<span class="normal">3</span></pre></div></td><td class="code"><div><pre><span></span><code>make_emperor.py -i wf_bdiv_even394/unweighted_unifrac_pc.txt -m sutton_mapping_file.txt –b &quot;Source,Condition,Source&amp;&amp;Condition&quot; –o emporer_output_unweighted

ls –l emporer_output_unweighted
</code></pre></div></td></tr></table></div>

<p>Run the last make_emporer.py command with the weighted_unifrac, using
the same command. Make sure you rename the output directory to
“emporer_output_weighted“. Look in the Emporer output directories and
note the index.html file outputs.</p>
<p>To view the index.html files created from the ‘make_emperor.py’ steps
above a browser like Firefox or Chrome can be used.</p>
<p>Important - To view the Emporer plots we need to switch from the VM to
your local machine. Open the Firefox browser and view the pre-computed
Emporer directories available at the following URL.
<a href="http://www.ebi.ac.uk/~sterk/emperor/">http://www.ebi.ac.uk/~sterk/emperor/</a></p>
<p>View the unweighted analysis on the Sutton sub data set:</p>
<p><a href="http://www.ebi.ac.uk/~sterk/emperor/sutton_subset_emporer_output_unweighted/">http://www.ebi.ac.uk/~sterk/emperor/sutton_subset_emporer_output_unweighted/</a></p>
<p>Check the ‘Colors’ options to view your samples. What can you determine
from the PCoA plot?</p>
<p>To change the colouring scheme click the ‘Colors’ tab. The available
metadata categories are sorted in alphabetical order. Extra information
on the features can be found at
<a href="http://qiime.org/emperor/description_index.html">http://qiime.org/emperor/description_index.html</a> Try changing the
scaling, visibility, label etc. to improve the display. Note that
Emperor generates publication quality figures in scalable vector
graphics (SVG) format.</p>
<p>Now look at the weighted unifrac analysis,
<a href="http://www.ebi.ac.uk/~sterk/emperor/sutton_subset_emporer_output_weighted/">http://www.ebi.ac.uk/~sterk/emperor/sutton_subset_emporer_output_weighted/</a></p>
<p>You can compare the sub data analysis against the full-denoised dataset.
<ttp://www.ebi.ac.uk/~sterk/emperor/sutton_full_denoised/></p>
<p>What are the main differences between the weighted and unweighted plots?</p>
<p>Did you notice any differences between the analysis on the sub data set
and the full denoised?</p>
<p>What are your final conclusions on the Sutton dataset?</p>
<p>What did each of the visualisation methods describe?</p>
<p>Taxonomy summary plots:</p>
<p>Alpha diversity plots:</p>
<p>Beta diversity PCoA plots:</p>
<p>Now that you are familiar with Emperor we will next view the forensic
analysis of the Fierer et. al. data. This experiment matches the
bacteria on surfaces to the skin-associated bacteria of the individual
who touched the surface. This study shows that skin-associated bacteria
can be readily recovered from surfaces (including single computer keys
and computer mice) and that the structure of these communities can be
used to differentiate objects handled by different individuals, even if
those objects have been left untouched for up to 2 weeks at room
temperature.</p>
<p>Please look the information contained in the mapping file and identify
those fields that can be plotted.</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span></pre></div></td><td class="code"><div><pre><span></span><code>cd ~/Desktop/Taxonomy/fierer/
less –S fierer_mapping_file.txt
</code></pre></div></td></tr></table></div>

<p>List any fields that can be plotted for this study. What is the
remainder of the information for?</p>
<p>Run Emperor to prepare the PCoA plots for visualisation</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>make_emperor.py -i fierer_unweighted_unifrac_pc.txt -m fierer_mapping_file.txt -b \ &quot;HOST_SUBJECT_ID,ENV_FEATURE,HOST_SUBJECT_ID&amp;&amp;ENV_FEATURE” -o emporer_output_unweighted
</code></pre></div></td></tr></table></div>

<p>Go to
<a href="http://www.ebi.ac.uk/~sterk/emperor/fierer_emporer_output_unweighted/">http://www.ebi.ac.uk/~sterk/emperor/fierer_emporer_output_unweighted/</a></p>
<p>What did you observe about the clustering?</p>
<p>What fields have been selected?</p>
<p>The on-line Qiime tutorials are very good and growing in numbers. It is
highly recommended checking out in future the many analysis options.</p>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      &copy; The authors; 2017-
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../../..", "features": [], "search": "../../../assets/javascripts/workers/search.12658920.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.5cf534bf.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
  </body>
</html>