
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
        <meta name="author" content="Authors">
      
      
      
        <link rel="prev" href="../../denovo-module-euk/eukaryotic_assem/">
      
      
        <link rel="next" href="../../metagenomics-module-qc/ngs-qc/">
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.2">
    
    
      
        <title>Scaffolding - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.f56500e0.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#key-learning-outcomes" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-header__button md-logo" aria-label="Markdown Manuals for Bioinformatics Training Platform (BTP)" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Markdown Manuals for Bioinformatics Training Platform (BTP)
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Scaffolding
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-nav__button md-logo" aria-label="Markdown Manuals for Bioinformatics Training Platform (BTP)" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../trainers/trainers/" class="md-nav__link">
        The Trainers
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          Timetable
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Timetable" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Timetable
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_introNGS/" class="md-nav__link">
        Intro to NGS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_cancer/" class="md-nav__link">
        Cancer Genomics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_longRead/" class="md-nav__link">
        Long Read
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../preamble/" class="md-nav__link">
        Workshop Information
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Introduction to NGS Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Introduction to NGS Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Introduction to NGS Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-cli/commandline/" class="md-nav__link">
        Introduction to Command Line
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-qc/ngs-qc/" class="md-nav__link">
        Data Quality
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-mapping/alignment/" class="md-nav__link">
        Read Alignment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-chip-seq/chip-seq/" class="md-nav__link">
        ChIP-seq
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-rna-seq/rna-seq/" class="md-nav__link">
        RNA-Seq
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-ngs-bio/biol/" class="md-nav__link">
        NGS Biological Insight
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-velvet/velvet/" class="md-nav__link">
        de novo Genome Assembly
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../btp-module-denovo-canu/denovo_canu/" class="md-nav__link">
        DeNovo Canu
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" >
      
      
      
      
        <label class="md-nav__link" for="__nav_6">
          Cancer Genomics Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Cancer Genomics Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Cancer Genomics Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-alignment/alignment/" class="md-nav__link">
        Read Alignment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-snv/snv/" class="md-nav__link">
        Single Nucleotide Variant Calling and Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-cnv/cnv-tut/" class="md-nav__link">
        Copy Number Variation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-sv/sv_tut/" class="md-nav__link">
        Structural Variant Analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-viz/visu/" class="md-nav__link">
        Variant Visualisation
      </a>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6_6" type="checkbox" id="__nav_6_6" >
      
      
      
      
        <label class="md-nav__link" for="__nav_6_6">
          Somatic Mutations
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Somatic Mutations" data-md-level="2">
        <label class="md-nav__title" for="__nav_6_6">
          <span class="md-nav__icon md-icon"></span>
          Somatic Mutations
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-somatic/01_signatures/" class="md-nav__link">
        Assessing somatic mutational signatures
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-somatic/02_intogen/" class="md-nav__link">
        Cohort analysis for the identification of driver genes
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_7" type="checkbox" id="__nav_7" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_7">
          DeNovo Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="DeNovo Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          DeNovo Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../denovo-module-bac/bac_asm/" class="md-nav__link">
        Bacterial Assembly
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../denovo-module-euk/eukaryotic_assem/" class="md-nav__link">
        Eukaryote Assembly
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Scaffolding
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Scaffolding
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav" aria-label="Resources You’ll be Using">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#author-information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#scaffolding-with-long-mate-pair-libraries-chalara-scaffolding-using-lmp" class="md-nav__link">
    Scaffolding with long mate pair libraries: Chalara scaffolding using LMP
  </a>
  
    <nav class="md-nav" aria-label="Scaffolding with long mate pair libraries: Chalara scaffolding using LMP">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#pair-end-assembly-using-chalara-dataset" class="md-nav__link">
    Pair-end assembly using Chalara dataset
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#chalara-scaffolding-using-long-mate-pair-reads-lmp" class="md-nav__link">
    Chalara scaffolding using long mate-pair reads (LMP)
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#chalara-beyond-first-pass" class="md-nav__link">
    Chalara: beyond first pass
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    References
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_8" type="checkbox" id="__nav_8" >
      
      
      
      
        <label class="md-nav__link" for="__nav_8">
          Metagenomics Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Metagenomics Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_8">
          <span class="md-nav__icon md-icon"></span>
          Metagenomics Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-qc/ngs-qc/" class="md-nav__link">
        Metagenomics QC
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-tax/tax/" class="md-nav__link">
        Metagenomics Taxonomic
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-vis/vis/" class="md-nav__link">
        Metagenomics Visualisation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-wga/wga/" class="md-nav__link">
        Metagenomics WGA
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../metagenomics-module-fda/fda/" class="md-nav__link">
        Metagenomics Functional Analysis
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav" aria-label="Resources You’ll be Using">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#author-information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#scaffolding-with-long-mate-pair-libraries-chalara-scaffolding-using-lmp" class="md-nav__link">
    Scaffolding with long mate pair libraries: Chalara scaffolding using LMP
  </a>
  
    <nav class="md-nav" aria-label="Scaffolding with long mate pair libraries: Chalara scaffolding using LMP">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#pair-end-assembly-using-chalara-dataset" class="md-nav__link">
    Pair-end assembly using Chalara dataset
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#chalara-scaffolding-using-long-mate-pair-reads-lmp" class="md-nav__link">
    Chalara scaffolding using long mate-pair reads (LMP)
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#chalara-beyond-first-pass" class="md-nav__link">
    Chalara: beyond first pass
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    References
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



  <h1>Scaffolding</h1>

<h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<p>After completing this module the trainee should be able to:</p>
<ul>
<li>
<p>Run an assembly using long mate pair (LMP) reads.</p>
</li>
<li>
<p>Compare results of LMP assemblies with those from the short reads.</p>
</li>
<li>
<p>Run scaffolding and assess the assembly statistics.</p>
</li>
</ul>
<hr />
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>Kmer Analysis Tool kit:<br />
<a href="https://github.com/TGAC/KAT">https://github.com/TGAC/KAT</a></p>
<p>Nextclip:<br />
<a href="https://github.com/richardmleggett/nextclip">https://github.com/richardmleggett/nextclip</a></p>
<p>Abyss:<br />
<a href="http://www.bcgsc.ca/platform/bioinfo/software/abyss">http://www.bcgsc.ca/platform/bioinfo/software/abyss</a></p>
<p>Soap Denovo:<br />
<a href="http://soap.genomics.org.cn/soapdenovo.html">http://soap.genomics.org.cn/soapdenovo.html</a></p>
<p>SOAPec:<br />
<a href="http://soap.genomics.org.cn/about.html">http://soap.genomics.org.cn/about.html</a></p>
<hr />
<h2 id="author-information">Author Information<a class="headerlink" href="#author-information" title="Permanent link">&para;</a></h2>
<p><em>Primary Author(s):</em><br />
Bernardo Clavijo, TGAC <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#66;&#101;&#114;&#110;&#97;&#114;&#100;&#111;&#46;&#67;&#108;&#97;&#118;&#105;&#106;&#111;&#64;&#116;&#103;&#97;&#99;&#46;&#97;&#99;&#46;&#117;&#107;">&#66;&#101;&#114;&#110;&#97;&#114;&#100;&#111;&#46;&#67;&#108;&#97;&#118;&#105;&#106;&#111;&#64;&#116;&#103;&#97;&#99;&#46;&#97;&#99;&#46;&#117;&#107;</a></p>
<p><em>Contributor(s):</em><br />
Sonika Tyagi, AGRF <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#115;&#111;&#110;&#105;&#107;&#97;&#46;&#116;&#121;&#97;&#103;&#105;&#64;&#97;&#103;&#114;&#102;&#46;&#111;&#114;&#103;&#46;&#97;&#117;">&#115;&#111;&#110;&#105;&#107;&#97;&#46;&#116;&#121;&#97;&#103;&#105;&#64;&#97;&#103;&#114;&#102;&#46;&#111;&#114;&#103;&#46;&#97;&#117;</a><br />
Zhiliang Chan <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#122;&#104;&#105;&#108;&#105;&#97;&#110;&#103;&#64;&#117;&#110;&#115;&#119;&#46;&#101;&#100;&#117;&#46;&#97;&#117;">&#122;&#104;&#105;&#108;&#105;&#97;&#110;&#103;&#64;&#117;&#110;&#115;&#119;&#46;&#101;&#100;&#117;&#46;&#97;&#117;</a></p>
<hr />
<h2 id="scaffolding-with-long-mate-pair-libraries-chalara-scaffolding-using-lmp">Scaffolding with long mate pair libraries: Chalara scaffolding using LMP<a class="headerlink" href="#scaffolding-with-long-mate-pair-libraries-chalara-scaffolding-using-lmp" title="Permanent link">&para;</a></h2>
<p>Scaffold is made of two or more contigs joined together using the read
pair information. In the absence of a high-quality reference genome, new
genome assemblies are often evaluated on the basis of the number of
scaffolds and contigs required to represent the genome. Other criteria
such as the alignment of reads back to assemblies, N50, and the length
of contigs and scaffolds relative to the size of the genome can also be
used to assess the quality of assemblies. In this exercise we will be
using a long mate pair data to perform assembly and scaffolding and we
will focus on how using LMP reads can affect the assemblies. Most part
of this tutorial has been precomputed and made available to you in
interest of time. We will spend more time exploring and discussing the
results.</p>
<h3 id="pair-end-assembly-using-chalara-dataset">Pair-end assembly using <em>Chalara</em> dataset<a class="headerlink" href="#pair-end-assembly-using-chalara-dataset" title="Permanent link">&para;</a></h3>
<p>Before doing a scaffolding, we need to build an assembly using the
pair-end reads first.</p>
<p>Let&rsquo;s go to the correct location where the files are:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>cd /home/trainee/scaffolding/cha_raw/test_1
</code></pre></div></td></tr></table></div>

<p><br></p>
<div class="admonition failure">
<p class="admonition-title">STOP</p>
<p>You DO NOT need to run this command. This has already been done for you.</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>abyss-pe name=cha1 k=27 in=&quot;../LIB2570_raw_R1.fastq ../LIB2570_raw_R2.fastq&quot; np=4
</code></pre></div></td></tr></table></div>

</div>
<p><br>
Let’s look at the stats by doing:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>less cha1-stats.tab
</code></pre></div></td></tr></table></div>

<p><br>
<strong>Table 1</strong>: Statistics of <em>Chalara</em> assembly by ABySS using k=27.</p>
<table>
<thead>
<tr>
<th><strong>n</strong></th>
<th><strong>n:500</strong></th>
<th><strong>L50</strong></th>
<th><strong>min</strong></th>
<th><strong>N80</strong></th>
<th><strong>N50</strong></th>
<th><strong>N20</strong></th>
<th><strong>E-size</strong></th>
<th><strong>max</strong></th>
<th><strong>sum</strong></th>
<th><strong>name</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>394789</td>
<td>11681</td>
<td>2094</td>
<td>500</td>
<td>2880</td>
<td>6254</td>
<td>12303</td>
<td>7936</td>
<td>44414</td>
<td>44.07e6</td>
<td>cha1-unitigs.fa</td>
</tr>
<tr>
<td>394199</td>
<td>11673</td>
<td>2097</td>
<td>500</td>
<td>2887</td>
<td>6255</td>
<td>12303</td>
<td>7937</td>
<td>44414</td>
<td>44.11e6</td>
<td>cha1-contigs.fa</td>
</tr>
<tr>
<td>394161</td>
<td>11647</td>
<td>2095</td>
<td>500</td>
<td>2898</td>
<td>6269</td>
<td>12303</td>
<td>7944</td>
<td>44414</td>
<td>44.12e6</td>
<td>cha1-scaffolds.fa</td>
</tr>
</tbody>
</table>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>cd /home/trainee/scaffolding/cha_raw/test_2
</code></pre></div></td></tr></table></div>

<p><br></p>
<div class="admonition failure">
<p class="admonition-title">STOP</p>
<p>This is also a pre-computed example for you:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code> abyss-pe name=cha2 k=61 in=&quot;../LIB2570_raw_R1.fastq ../LIB2570_raw_R2.fastq&quot; np=4
</code></pre></div></td></tr></table></div>

</div>
<p><br>
Again, we need to look at the statistics of our new assembly using k=61:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>less cha2-stats.tab
</code></pre></div></td></tr></table></div>

<p><br>
It should be looking like this:</p>
<p><strong>Table 2</strong>: Statistics of <em>Chalara</em> assembly by ABySS using k=61.</p>
<table>
<thead>
<tr>
<th><strong>n</strong></th>
<th><strong>n:500</strong></th>
<th><strong>L50</strong></th>
<th><strong>min</strong></th>
<th><strong>N80</strong></th>
<th><strong>N50</strong></th>
<th><strong>N20</strong></th>
<th><strong>E-size</strong></th>
<th><strong>max</strong></th>
<th><strong>sum</strong></th>
<th><strong>name</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>130547</td>
<td>12596</td>
<td>1770</td>
<td>500</td>
<td>3352</td>
<td>8379</td>
<td>16380</td>
<td>10518</td>
<td>54300</td>
<td>50.75e6</td>
<td>cha2-unitigs.fa</td>
</tr>
<tr>
<td>130210</td>
<td>12575</td>
<td>1771</td>
<td>500</td>
<td>3363</td>
<td>8382</td>
<td>16380</td>
<td>10525</td>
<td>54300</td>
<td>50.78e6</td>
<td>cha2-contigs.fa</td>
</tr>
<tr>
<td>130182</td>
<td>12555</td>
<td>1769</td>
<td>500</td>
<td>3377</td>
<td>8394</td>
<td>16380</td>
<td>10534</td>
<td>54300</td>
<td>50.78e6</td>
<td>cha2-scaffolds.fa</td>
</tr>
</tbody>
</table>
<p><br></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>The assembly contiguity is worse than <em>Fusarium</em>, why?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<ul>
<li>Genome characteristics.</li>
<li>Data</li>
<li>Coverage</li>
<li>Error distributions</li>
<li>Read sizes</li>
<li>Fragment sizes</li>
<li>Kmer spectra</li>
</ul>
</details>
</div>
</div>
<h3 id="chalara-scaffolding-using-long-mate-pair-reads-lmp">Chalara scaffolding using long mate-pair reads (LMP)<a class="headerlink" href="#chalara-scaffolding-using-long-mate-pair-reads-lmp" title="Permanent link">&para;</a></h3>
<p>Let’s put some LMP in there.</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>cd /home/trainee/scaffolding/cha_raw/test_3
</code></pre></div></td></tr></table></div>

<p><br></p>
<div class="admonition failure">
<p class="admonition-title">STOP</p>
<p>This is also a pre-computed example for you:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>abyss-pe name=chalmp1 k=61 in=&quot;../LIB2570_raw_R1.fastq ../LIB2570_raw_R2.fastq&quot; mp=&quot;lmp1&quot; lmp1=&quot;../LIB8209_raw_R1.fastq ../LIB8209_raw_R2.fastq&quot; np=4
</code></pre></div></td></tr></table></div>

</div>
<p><br>
  <div class="highlight"><pre><span></span><code>less chalmp1-stats.tab
</code></pre></div></p>
<p><br>
<strong>Table 3</strong>: Statistics of <em>Chalara</em> assembly by ABySS using k=27.</p>
<table>
<thead>
<tr>
<th><strong>n</strong></th>
<th><strong>n:500</strong></th>
<th><strong>L50</strong></th>
<th><strong>min</strong></th>
<th><strong>N80</strong></th>
<th><strong>N50</strong></th>
<th><strong>N20</strong></th>
<th><strong>E-size</strong></th>
<th><strong>max</strong></th>
<th><strong>sum</strong></th>
<th><strong>name</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>130548</td>
<td>12596</td>
<td>1770</td>
<td>500</td>
<td>3352</td>
<td>8379</td>
<td>16380</td>
<td>10518</td>
<td>54300</td>
<td>50.75e6</td>
<td>chalmp1-unitigs.fa</td>
</tr>
<tr>
<td>130211</td>
<td>12575</td>
<td>1771</td>
<td>500</td>
<td>3363</td>
<td>8382</td>
<td>16380</td>
<td>10525</td>
<td>54300</td>
<td>50.78e6</td>
<td>chalmp1-contigs.fa</td>
</tr>
<tr>
<td>130148</td>
<td>12545</td>
<td>1769</td>
<td>500</td>
<td>3380</td>
<td>8400</td>
<td>16380</td>
<td>10535</td>
<td>54300</td>
<td>50.78e6</td>
<td>chalmp1-scaffolds.fa</td>
</tr>
</tbody>
</table>
<p>So&hellip; what happened? Let’s check the data by:</p>
<ul>
<li>Kmer spectra</li>
<li>Fragment sizes</li>
</ul>
<p><br>
Any hints on the protocol? And a not-so-obvious property:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>kat plot spectra-mx --intersection -x 30 -y 15000000 -o pe_vs_lmp-main.mx.spectra-mx.png pe_vs_lmp-main.mx
</code></pre></div></td></tr></table></div>

<p><br>
Do you remember LMP required pre-processing? Let’s try with processed LMP.</p>
<p>Prior task (already made) preprocess the LMP with <code>nextclip</code>.</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>cd /home/trainee/scaffolding/cha_proc/test_1
</code></pre></div></td></tr></table></div>

<p><br></p>
<div class="admonition failure">
<p class="admonition-title">STOP</p>
<p>This is also a pre-computed example for you:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>abyss-pe name=chalmpproc1 k=61 in=&quot;../../cha_raw/LIB2570_raw_R1.fastq ../../cha_raw/LIB2570_raw_R2.fastq&quot; mp=&quot;proclmp1&quot; proclmp1=&quot;../LIB8209_preproc_R1.fastq ../LIB8209_preproc_R2.fastq&quot; np=4
</code></pre></div></td></tr></table></div>

</div>
<p><br></p>
<div class="highlight"><pre><span></span><code>less chalmpproc1-stats.tab
</code></pre></div>
<p><br>
<strong>Table 4</strong>: Statistics of <em>Chalara</em> assembly by ABySS using k=61 with LMP.</p>
<table>
<thead>
<tr>
<th><strong>n</strong></th>
<th><strong>n:500</strong></th>
<th><strong>L50</strong></th>
<th><strong>min</strong></th>
<th><strong>N80</strong></th>
<th><strong>N50</strong></th>
<th><strong>N20</strong></th>
<th><strong>E-size</strong></th>
<th><strong>max</strong></th>
<th><strong>sum</strong></th>
<th><strong>name</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>130548</td>
<td>12596</td>
<td>1770</td>
<td>500</td>
<td>3352</td>
<td>8379</td>
<td>16380</td>
<td>10518</td>
<td>54300</td>
<td>50.75e6</td>
<td>chalmpproc1-unitigs.fa</td>
</tr>
<tr>
<td>130211</td>
<td>12575</td>
<td>1771</td>
<td>500</td>
<td>3363</td>
<td>8382</td>
<td>16380</td>
<td>10525</td>
<td>54300</td>
<td>50.78e6</td>
<td>chalmpproc1-contigs.fa</td>
</tr>
<tr>
<td>122061</td>
<td>6679</td>
<td>167</td>
<td>500</td>
<td>18609</td>
<td>87510</td>
<td>187171</td>
<td>106178</td>
<td>397967</td>
<td>51.15e6</td>
<td>chalmpproc1-scaffolds.fa</td>
</tr>
</tbody>
</table>
<p><br>
That’s much better!</p>
<hr />
<h2 id="chalara-beyond-first-pass">Chalara: beyond first pass<a class="headerlink" href="#chalara-beyond-first-pass" title="Permanent link">&para;</a></h2>
<p>Do you remember these?</p>
<p>Genome characteristics. Data:</p>
<ul>
<li>Coverage</li>
<li>Error distributions</li>
<li>Read sizes</li>
<li>Fragment sizes</li>
<li>Kmer spectra</li>
</ul>
<p>Let’s think a bit and try to improve the assembly.</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>If you look at the pre-processed LMP, do you notice anything peculiar?</p>
</div>
<p>Testing the inclusion of heavily pre-processed LMP coverage into the DBG</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>cd /home/trainee/scaffolding/cha_proc/test_2
</code></pre></div></td></tr></table></div>

<p><br></p>
<div class="admonition failure">
<p class="admonition-title">STOP</p>
<p>This is also a pre-computed example for you:</p>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>abyss-pe name=chalmp2 k=61 se=&quot;../LIB8209_preproc_single.fastq&quot; lib=&quot;lmp1&quot; lmp1=&quot;../LIB8209_preproc_R1.fastq ../LIB8209_preproc_R2.fastq&quot; np=4 &gt;chaproclmp2.log 2&gt;&amp;1
</code></pre></div></td></tr></table></div>

</div>
<p><br></p>
<div class="highlight"><pre><span></span><code>less chalmp2-stats.tab
</code></pre></div>
<p><br>
Let’s look at the stats.  </p>
<p><strong>Table 5</strong>: Statistics of improved <em>Chalara</em> assembly by ABySS using k=61 with LMP.</p>
<table>
<thead>
<tr>
<th><strong>n</strong></th>
<th><strong>n:500</strong></th>
<th><strong>L50</strong></th>
<th><strong>min</strong></th>
<th><strong>N80</strong></th>
<th><strong>N50</strong></th>
<th><strong>N20</strong></th>
<th><strong>E-size</strong></th>
<th><strong>max</strong></th>
<th><strong>sum</strong></th>
<th><strong>name</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>128306</td>
<td>10150</td>
<td>1182</td>
<td>500</td>
<td>4954</td>
<td>12585</td>
<td>24777</td>
<td>15693</td>
<td>68684</td>
<td>51.03e6</td>
<td>chaproclmp4-unitigs.fa</td>
</tr>
<tr>
<td>118939</td>
<td>5772</td>
<td>362</td>
<td>500</td>
<td>15717</td>
<td>41870</td>
<td>82033</td>
<td>52819</td>
<td>252066</td>
<td>52.24e6</td>
<td>chaproclmp4-contigs.fa</td>
</tr>
<tr>
<td>116139</td>
<td>4014</td>
<td>141</td>
<td>500</td>
<td>41145</td>
<td>109273</td>
<td>224986</td>
<td>133450</td>
<td>460979</td>
<td>52.56e6</td>
<td>chaproclmp4-scaffolds.fa</td>
</tr>
</tbody>
</table>
<hr />
<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<ol>
<li>
<p>Robert Ekblom* and Jochen B. W. Wolf. &ldquo;A field guide to whole-genome
    sequencing, assembly and annotation&rdquo;. Evolutionary Applications
    Special Issue: Evolutionary Conservation Volume 7, Issue 9, pages
    1026 – 1042, November 2014</p>
</li>
<li>
<p>De novo genome assembly: what every biologist should know. Nature
    Methods 9, 333 – 337 (2012) doi:10.1038/nmeth.1935</p>
</li>
</ol>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      &copy; The authors; 2017-
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../../..", "features": [], "search": "../../../assets/javascripts/workers/search.12658920.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.5cf534bf.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
  </body>
</html>